// @imstar15/api-augment/interfaces/lookup.js
"use strict";
// Auto-generated via `yarn polkadot-types-from-defs`, do not edit
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
/* eslint-disable sort-keys */
exports.default = {
    /**
     * Lookup3: frame_system::AccountInfo<Index, pallet_balances::AccountData<Balance>>
     **/
    FrameSystemAccountInfo: {
        nonce: 'u32',
        consumers: 'u32',
        providers: 'u32',
        sufficients: 'u32',
        data: 'PalletBalancesAccountData'
    },
    /**
     * Lookup5: pallet_balances::AccountData<Balance>
     **/
    PalletBalancesAccountData: {
        free: 'u128',
        reserved: 'u128',
        miscFrozen: 'u128',
        feeFrozen: 'u128'
    },
    /**
     * Lookup7: frame_support::weights::PerDispatchClass<T>
     **/
    FrameSupportWeightsPerDispatchClassU64: {
        normal: 'u64',
        operational: 'u64',
        mandatory: 'u64'
    },
    /**
     * Lookup11: sp_runtime::generic::digest::Digest
     **/
    SpRuntimeDigest: {
        logs: 'Vec<SpRuntimeDigestDigestItem>'
    },
    /**
     * Lookup13: sp_runtime::generic::digest::DigestItem
     **/
    SpRuntimeDigestDigestItem: {
        _enum: {
            Other: 'Bytes',
            __Unused1: 'Null',
            __Unused2: 'Null',
            __Unused3: 'Null',
            Consensus: '([u8;4],Bytes)',
            Seal: '([u8;4],Bytes)',
            PreRuntime: '([u8;4],Bytes)',
            __Unused7: 'Null',
            RuntimeEnvironmentUpdated: 'Null'
        }
    },
    /**
     * Lookup16: frame_system::EventRecord<turing_runtime::Event, primitive_types::H256>
     **/
    FrameSystemEventRecord: {
        phase: 'FrameSystemPhase',
        event: 'Event',
        topics: 'Vec<H256>'
    },
    /**
     * Lookup18: frame_system::pallet::Event<T>
     **/
    FrameSystemEvent: {
        _enum: {
            ExtrinsicSuccess: {
                dispatchInfo: 'FrameSupportWeightsDispatchInfo',
            },
            ExtrinsicFailed: {
                dispatchError: 'SpRuntimeDispatchError',
                dispatchInfo: 'FrameSupportWeightsDispatchInfo',
            },
            CodeUpdated: 'Null',
            NewAccount: {
                account: 'AccountId32',
            },
            KilledAccount: {
                account: 'AccountId32',
            },
            Remarked: {
                _alias: {
                    hash_: 'hash',
                },
                sender: 'AccountId32',
                hash_: 'H256'
            }
        }
    },
    /**
     * Lookup19: frame_support::weights::DispatchInfo
     **/
    FrameSupportWeightsDispatchInfo: {
        weight: 'u64',
        class: 'FrameSupportWeightsDispatchClass',
        paysFee: 'FrameSupportWeightsPays'
    },
    /**
     * Lookup20: frame_support::weights::DispatchClass
     **/
    FrameSupportWeightsDispatchClass: {
        _enum: ['Normal', 'Operational', 'Mandatory']
    },
    /**
     * Lookup21: frame_support::weights::Pays
     **/
    FrameSupportWeightsPays: {
        _enum: ['Yes', 'No']
    },
    /**
     * Lookup22: sp_runtime::DispatchError
     **/
    SpRuntimeDispatchError: {
        _enum: {
            Other: 'Null',
            CannotLookup: 'Null',
            BadOrigin: 'Null',
            Module: 'SpRuntimeModuleError',
            ConsumerRemaining: 'Null',
            NoProviders: 'Null',
            TooManyConsumers: 'Null',
            Token: 'SpRuntimeTokenError',
            Arithmetic: 'SpRuntimeArithmeticError',
            Transactional: 'SpRuntimeTransactionalError'
        }
    },
    /**
     * Lookup23: sp_runtime::ModuleError
     **/
    SpRuntimeModuleError: {
        index: 'u8',
        error: '[u8;4]'
    },
    /**
     * Lookup24: sp_runtime::TokenError
     **/
    SpRuntimeTokenError: {
        _enum: ['NoFunds', 'WouldDie', 'BelowMinimum', 'CannotCreate', 'UnknownAsset', 'Frozen', 'Unsupported']
    },
    /**
     * Lookup25: sp_runtime::ArithmeticError
     **/
    SpRuntimeArithmeticError: {
        _enum: ['Underflow', 'Overflow', 'DivisionByZero']
    },
    /**
     * Lookup26: sp_runtime::TransactionalError
     **/
    SpRuntimeTransactionalError: {
        _enum: ['LimitReached', 'NoLayer']
    },
    /**
     * Lookup27: cumulus_pallet_parachain_system::pallet::Event<T>
     **/
    CumulusPalletParachainSystemEvent: {
        _enum: {
            ValidationFunctionStored: 'Null',
            ValidationFunctionApplied: {
                relayChainBlockNum: 'u32',
            },
            ValidationFunctionDiscarded: 'Null',
            UpgradeAuthorized: {
                codeHash: 'H256',
            },
            DownwardMessagesReceived: {
                count: 'u32',
            },
            DownwardMessagesProcessed: {
                weightUsed: 'u64',
                dmqHead: 'H256'
            }
        }
    },
    /**
     * Lookup28: pallet_balances::pallet::Event<T, I>
     **/
    PalletBalancesEvent: {
        _enum: {
            Endowed: {
                account: 'AccountId32',
                freeBalance: 'u128',
            },
            DustLost: {
                account: 'AccountId32',
                amount: 'u128',
            },
            Transfer: {
                from: 'AccountId32',
                to: 'AccountId32',
                amount: 'u128',
            },
            BalanceSet: {
                who: 'AccountId32',
                free: 'u128',
                reserved: 'u128',
            },
            Reserved: {
                who: 'AccountId32',
                amount: 'u128',
            },
            Unreserved: {
                who: 'AccountId32',
                amount: 'u128',
            },
            ReserveRepatriated: {
                from: 'AccountId32',
                to: 'AccountId32',
                amount: 'u128',
                destinationStatus: 'FrameSupportTokensMiscBalanceStatus',
            },
            Deposit: {
                who: 'AccountId32',
                amount: 'u128',
            },
            Withdraw: {
                who: 'AccountId32',
                amount: 'u128',
            },
            Slashed: {
                who: 'AccountId32',
                amount: 'u128'
            }
        }
    },
    /**
     * Lookup29: frame_support::traits::tokens::misc::BalanceStatus
     **/
    FrameSupportTokensMiscBalanceStatus: {
        _enum: ['Free', 'Reserved']
    },
    /**
     * Lookup30: orml_tokens::module::Event<T>
     **/
    OrmlTokensModuleEvent: {
        _enum: {
            Endowed: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            DustLost: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            Transfer: {
                currencyId: 'u32',
                from: 'AccountId32',
                to: 'AccountId32',
                amount: 'u128',
            },
            Reserved: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            Unreserved: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            ReserveRepatriated: {
                currencyId: 'u32',
                from: 'AccountId32',
                to: 'AccountId32',
                amount: 'u128',
                status: 'FrameSupportTokensMiscBalanceStatus',
            },
            BalanceSet: {
                currencyId: 'u32',
                who: 'AccountId32',
                free: 'u128',
                reserved: 'u128',
            },
            TotalIssuanceSet: {
                currencyId: 'u32',
                amount: 'u128',
            },
            Withdrawn: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            Slashed: {
                currencyId: 'u32',
                who: 'AccountId32',
                freeAmount: 'u128',
                reservedAmount: 'u128',
            },
            Deposited: {
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            LockSet: {
                lockId: '[u8;8]',
                currencyId: 'u32',
                who: 'AccountId32',
                amount: 'u128',
            },
            LockRemoved: {
                lockId: '[u8;8]',
                currencyId: 'u32',
                who: 'AccountId32'
            }
        }
    },
    /**
     * Lookup32: pallet_transaction_payment::pallet::Event<T>
     **/
    PalletTransactionPaymentEvent: {
        _enum: {
            TransactionFeePaid: {
                who: 'AccountId32',
                actualFee: 'u128',
                tip: 'u128'
            }
        }
    },
    /**
     * Lookup33: orml_asset_registry::module::Event<T>
     **/
    OrmlAssetRegistryModuleEvent: {
        _enum: {
            RegisteredAsset: {
                assetId: 'u32',
                metadata: 'OrmlTraitsAssetRegistryAssetMetadata',
            },
            UpdatedAsset: {
                assetId: 'u32',
                metadata: 'OrmlTraitsAssetRegistryAssetMetadata'
            }
        }
    },
    /**
     * Lookup34: orml_traits::asset_registry::AssetMetadata<Balance, primitives::assets::CustomMetadata>
     **/
    OrmlTraitsAssetRegistryAssetMetadata: {
        decimals: 'u32',
        name: 'Bytes',
        symbol: 'Bytes',
        existentialDeposit: 'u128',
        location: 'Option<XcmVersionedMultiLocation>',
        additional: 'PrimitivesAssetsCustomMetadata'
    },
    /**
     * Lookup35: primitives::assets::CustomMetadata
     **/
    PrimitivesAssetsCustomMetadata: {
        feePerSecond: 'Option<u128>',
        conversionRate: 'Option<PrimitivesAssetsConversionRate>'
    },
    /**
     * Lookup38: primitives::assets::ConversionRate
     **/
    PrimitivesAssetsConversionRate: {
        native: 'u32',
        foreign: 'u32'
    },
    /**
     * Lookup40: xcm::VersionedMultiLocation
     **/
    XcmVersionedMultiLocation: {
        _enum: {
            V0: 'XcmV0MultiLocation',
            V1: 'XcmV1MultiLocation'
        }
    },
    /**
     * Lookup41: xcm::v0::multi_location::MultiLocation
     **/
    XcmV0MultiLocation: {
        _enum: {
            Null: 'Null',
            X1: 'XcmV0Junction',
            X2: '(XcmV0Junction,XcmV0Junction)',
            X3: '(XcmV0Junction,XcmV0Junction,XcmV0Junction)',
            X4: '(XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction)',
            X5: '(XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction)',
            X6: '(XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction)',
            X7: '(XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction)',
            X8: '(XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction,XcmV0Junction)'
        }
    },
    /**
     * Lookup42: xcm::v0::junction::Junction
     **/
    XcmV0Junction: {
        _enum: {
            Parent: 'Null',
            Parachain: 'Compact<u32>',
            AccountId32: {
                network: 'XcmV0JunctionNetworkId',
                id: '[u8;32]',
            },
            AccountIndex64: {
                network: 'XcmV0JunctionNetworkId',
                index: 'Compact<u64>',
            },
            AccountKey20: {
                network: 'XcmV0JunctionNetworkId',
                key: '[u8;20]',
            },
            PalletInstance: 'u8',
            GeneralIndex: 'Compact<u128>',
            GeneralKey: 'Bytes',
            OnlyChild: 'Null',
            Plurality: {
                id: 'XcmV0JunctionBodyId',
                part: 'XcmV0JunctionBodyPart'
            }
        }
    },
    /**
     * Lookup44: xcm::v0::junction::NetworkId
     **/
    XcmV0JunctionNetworkId: {
        _enum: {
            Any: 'Null',
            Named: 'Bytes',
            Polkadot: 'Null',
            Kusama: 'Null'
        }
    },
    /**
     * Lookup49: xcm::v0::junction::BodyId
     **/
    XcmV0JunctionBodyId: {
        _enum: {
            Unit: 'Null',
            Named: 'Bytes',
            Index: 'Compact<u32>',
            Executive: 'Null',
            Technical: 'Null',
            Legislative: 'Null',
            Judicial: 'Null'
        }
    },
    /**
     * Lookup50: xcm::v0::junction::BodyPart
     **/
    XcmV0JunctionBodyPart: {
        _enum: {
            Voice: 'Null',
            Members: {
                count: 'Compact<u32>',
            },
            Fraction: {
                nom: 'Compact<u32>',
                denom: 'Compact<u32>',
            },
            AtLeastProportion: {
                nom: 'Compact<u32>',
                denom: 'Compact<u32>',
            },
            MoreThanProportion: {
                nom: 'Compact<u32>',
                denom: 'Compact<u32>'
            }
        }
    },
    /**
     * Lookup51: xcm::v1::multilocation::MultiLocation
     **/
    XcmV1MultiLocation: {
        parents: 'u8',
        interior: 'XcmV1MultilocationJunctions'
    },
    /**
     * Lookup52: xcm::v1::multilocation::Junctions
     **/
    XcmV1MultilocationJunctions: {
        _enum: {
            Here: 'Null',
            X1: 'XcmV1Junction',
            X2: '(XcmV1Junction,XcmV1Junction)',
            X3: '(XcmV1Junction,XcmV1Junction,XcmV1Junction)',
            X4: '(XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction)',
            X5: '(XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction)',
            X6: '(XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction)',
            X7: '(XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction)',
            X8: '(XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction,XcmV1Junction)'
        }
    },
    /**
     * Lookup53: xcm::v1::junction::Junction
     **/
    XcmV1Junction: {
        _enum: {
            Parachain: 'Compact<u32>',
            AccountId32: {
                network: 'XcmV0JunctionNetworkId',
                id: '[u8;32]',
            },
            AccountIndex64: {
                network: 'XcmV0JunctionNetworkId',
                index: 'Compact<u64>',
            },
            AccountKey20: {
                network: 'XcmV0JunctionNetworkId',
                key: '[u8;20]',
            },
            PalletInstance: 'u8',
            GeneralIndex: 'Compact<u128>',
            GeneralKey: 'Bytes',
            OnlyChild: 'Null',
            Plurality: {
                id: 'XcmV0JunctionBodyId',
                part: 'XcmV0JunctionBodyPart'
            }
        }
    },
    /**
     * Lookup54: pallet_session::pallet::Event
     **/
    PalletSessionEvent: {
        _enum: {
            NewSession: {
                sessionIndex: 'u32'
            }
        }
    },
    /**
     * Lookup55: pallet_parachain_staking::pallet::Event<T>
     **/
    PalletParachainStakingEvent: {
        _enum: {
            NewRound: {
                startingBlock: 'u32',
                round: 'u32',
                selectedCollatorsNumber: 'u32',
                totalBalance: 'u128',
            },
            JoinedCollatorCandidates: {
                account: 'AccountId32',
                amountLocked: 'u128',
                newTotalAmtLocked: 'u128',
            },
            CollatorChosen: {
                round: 'u32',
                collatorAccount: 'AccountId32',
                totalExposedAmount: 'u128',
            },
            CandidateBondLessRequested: {
                candidate: 'AccountId32',
                amountToDecrease: 'u128',
                executeRound: 'u32',
            },
            CandidateBondedMore: {
                candidate: 'AccountId32',
                amount: 'u128',
                newTotalBond: 'u128',
            },
            CandidateBondedLess: {
                candidate: 'AccountId32',
                amount: 'u128',
                newBond: 'u128',
            },
            CandidateWentOffline: {
                candidate: 'AccountId32',
            },
            CandidateBackOnline: {
                candidate: 'AccountId32',
            },
            CandidateScheduledExit: {
                exitAllowedRound: 'u32',
                candidate: 'AccountId32',
                scheduledExit: 'u32',
            },
            CancelledCandidateExit: {
                candidate: 'AccountId32',
            },
            CancelledCandidateBondLess: {
                candidate: 'AccountId32',
                amount: 'u128',
                executeRound: 'u32',
            },
            CandidateLeft: {
                exCandidate: 'AccountId32',
                unlockedAmount: 'u128',
                newTotalAmtLocked: 'u128',
            },
            DelegationDecreaseScheduled: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                amountToDecrease: 'u128',
                executeRound: 'u32',
            },
            DelegationIncreased: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                amount: 'u128',
                inTop: 'bool',
            },
            DelegationDecreased: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                amount: 'u128',
                inTop: 'bool',
            },
            DelegatorExitScheduled: {
                round: 'u32',
                delegator: 'AccountId32',
                scheduledExit: 'u32',
            },
            DelegationRevocationScheduled: {
                round: 'u32',
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                scheduledExit: 'u32',
            },
            DelegatorLeft: {
                delegator: 'AccountId32',
                unstakedAmount: 'u128',
            },
            DelegationRevoked: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                unstakedAmount: 'u128',
            },
            DelegationKicked: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                unstakedAmount: 'u128',
            },
            DelegatorExitCancelled: {
                delegator: 'AccountId32',
            },
            CancelledDelegationRequest: {
                delegator: 'AccountId32',
                cancelledRequest: 'PalletParachainStakingDelegationRequestsCancelledScheduledRequest',
                collator: 'AccountId32',
            },
            Delegation: {
                delegator: 'AccountId32',
                lockedAmount: 'u128',
                candidate: 'AccountId32',
                delegatorPosition: 'PalletParachainStakingDelegatorAdded',
            },
            DelegatorLeftCandidate: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
                unstakedAmount: 'u128',
                totalCandidateStaked: 'u128',
            },
            Rewarded: {
                account: 'AccountId32',
                rewards: 'u128',
            },
            ReservedForParachainBond: {
                account: 'AccountId32',
                value: 'u128',
            },
            ParachainBondAccountSet: {
                _alias: {
                    new_: 'new',
                },
                old: 'AccountId32',
                new_: 'AccountId32',
            },
            ParachainBondReservePercentSet: {
                _alias: {
                    new_: 'new',
                },
                old: 'Percent',
                new_: 'Percent',
            },
            InflationSet: {
                annualMin: 'Perbill',
                annualIdeal: 'Perbill',
                annualMax: 'Perbill',
                roundMin: 'Perbill',
                roundIdeal: 'Perbill',
                roundMax: 'Perbill',
            },
            StakeExpectationsSet: {
                expectMin: 'u128',
                expectIdeal: 'u128',
                expectMax: 'u128',
            },
            TotalSelectedSet: {
                _alias: {
                    new_: 'new',
                },
                old: 'u32',
                new_: 'u32',
            },
            CollatorCommissionSet: {
                _alias: {
                    new_: 'new',
                },
                old: 'Perbill',
                new_: 'Perbill',
            },
            BlocksPerRoundSet: {
                _alias: {
                    new_: 'new',
                },
                currentRound: 'u32',
                firstBlock: 'u32',
                old: 'u32',
                new_: 'u32',
                newPerRoundInflationMin: 'Perbill',
                newPerRoundInflationIdeal: 'Perbill',
                newPerRoundInflationMax: 'Perbill'
            }
        }
    },
    /**
     * Lookup57: pallet_parachain_staking::delegation_requests::CancelledScheduledRequest<Balance>
     **/
    PalletParachainStakingDelegationRequestsCancelledScheduledRequest: {
        whenExecutable: 'u32',
        action: 'PalletParachainStakingDelegationRequestsDelegationAction'
    },
    /**
     * Lookup58: pallet_parachain_staking::delegation_requests::DelegationAction<Balance>
     **/
    PalletParachainStakingDelegationRequestsDelegationAction: {
        _enum: {
            Revoke: 'u128',
            Decrease: 'u128'
        }
    },
    /**
     * Lookup59: pallet_parachain_staking::types::DelegatorAdded<B>
     **/
    PalletParachainStakingDelegatorAdded: {
        _enum: {
            AddedToTop: {
                newTotal: 'u128',
            },
            AddedToBottom: 'Null'
        }
    },
    /**
     * Lookup62: pallet_valve::pallet::Event<T>
     **/
    PalletValveEvent: {
        _enum: {
            ValveClosed: 'Null',
            ValveOpen: 'Null',
            PalletGateClosed: {
                palletNameBytes: 'Bytes',
            },
            PalletGateOpen: {
                palletNameBytes: 'Bytes',
            },
            ScheduledTasksStopped: 'Null',
            ScheduledTasksResumed: 'Null',
            PalletGatesClosed: {
                count: 'u8'
            }
        }
    },
    /**
     * Lookup63: pallet_identity::pallet::Event<T>
     **/
    PalletIdentityEvent: {
        _enum: {
            IdentitySet: {
                who: 'AccountId32',
            },
            IdentityCleared: {
                who: 'AccountId32',
                deposit: 'u128',
            },
            IdentityKilled: {
                who: 'AccountId32',
                deposit: 'u128',
            },
            JudgementRequested: {
                who: 'AccountId32',
                registrarIndex: 'u32',
            },
            JudgementUnrequested: {
                who: 'AccountId32',
                registrarIndex: 'u32',
            },
            JudgementGiven: {
                target: 'AccountId32',
                registrarIndex: 'u32',
            },
            RegistrarAdded: {
                registrarIndex: 'u32',
            },
            SubIdentityAdded: {
                sub: 'AccountId32',
                main: 'AccountId32',
                deposit: 'u128',
            },
            SubIdentityRemoved: {
                sub: 'AccountId32',
                main: 'AccountId32',
                deposit: 'u128',
            },
            SubIdentityRevoked: {
                sub: 'AccountId32',
                main: 'AccountId32',
                deposit: 'u128'
            }
        }
    },
    /**
     * Lookup64: pallet_proxy::pallet::Event<T>
     **/
    PalletProxyEvent: {
        _enum: {
            ProxyExecuted: {
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            AnonymousCreated: {
                anonymous: 'AccountId32',
                who: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                disambiguationIndex: 'u16',
            },
            Announced: {
                real: 'AccountId32',
                proxy: 'AccountId32',
                callHash: 'H256',
            },
            ProxyAdded: {
                delegator: 'AccountId32',
                delegatee: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                delay: 'u32',
            },
            ProxyRemoved: {
                delegator: 'AccountId32',
                delegatee: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                delay: 'u32'
            }
        }
    },
    /**
     * Lookup67: turing_runtime::ProxyType
     **/
    TuringRuntimeProxyType: {
        _enum: ['Any', 'Session', 'Staking']
    },
    /**
     * Lookup69: pallet_utility::pallet::Event
     **/
    PalletUtilityEvent: {
        _enum: {
            BatchInterrupted: {
                index: 'u32',
                error: 'SpRuntimeDispatchError',
            },
            BatchCompleted: 'Null',
            BatchCompletedWithErrors: 'Null',
            ItemCompleted: 'Null',
            ItemFailed: {
                error: 'SpRuntimeDispatchError',
            },
            DispatchedAs: {
                result: 'Result<Null, SpRuntimeDispatchError>'
            }
        }
    },
    /**
     * Lookup70: cumulus_pallet_xcmp_queue::pallet::Event<T>
     **/
    CumulusPalletXcmpQueueEvent: {
        _enum: {
            Success: {
                messageHash: 'Option<H256>',
                weight: 'u64',
            },
            Fail: {
                messageHash: 'Option<H256>',
                error: 'XcmV2TraitsError',
                weight: 'u64',
            },
            BadVersion: {
                messageHash: 'Option<H256>',
            },
            BadFormat: {
                messageHash: 'Option<H256>',
            },
            UpwardMessageSent: {
                messageHash: 'Option<H256>',
            },
            XcmpMessageSent: {
                messageHash: 'Option<H256>',
            },
            OverweightEnqueued: {
                sender: 'u32',
                sentAt: 'u32',
                index: 'u64',
                required: 'u64',
            },
            OverweightServiced: {
                index: 'u64',
                used: 'u64'
            }
        }
    },
    /**
     * Lookup72: xcm::v2::traits::Error
     **/
    XcmV2TraitsError: {
        _enum: {
            Overflow: 'Null',
            Unimplemented: 'Null',
            UntrustedReserveLocation: 'Null',
            UntrustedTeleportLocation: 'Null',
            MultiLocationFull: 'Null',
            MultiLocationNotInvertible: 'Null',
            BadOrigin: 'Null',
            InvalidLocation: 'Null',
            AssetNotFound: 'Null',
            FailedToTransactAsset: 'Null',
            NotWithdrawable: 'Null',
            LocationCannotHold: 'Null',
            ExceedsMaxMessageSize: 'Null',
            DestinationUnsupported: 'Null',
            Transport: 'Null',
            Unroutable: 'Null',
            UnknownClaim: 'Null',
            FailedToDecode: 'Null',
            MaxWeightInvalid: 'Null',
            NotHoldingFees: 'Null',
            TooExpensive: 'Null',
            Trap: 'u64',
            UnhandledXcmVersion: 'Null',
            WeightLimitReached: 'u64',
            Barrier: 'Null',
            WeightNotComputable: 'Null'
        }
    },
    /**
     * Lookup74: pallet_xcm::pallet::Event<T>
     **/
    PalletXcmEvent: {
        _enum: {
            Attempted: 'XcmV2TraitsOutcome',
            Sent: '(XcmV1MultiLocation,XcmV1MultiLocation,XcmV2Xcm)',
            UnexpectedResponse: '(XcmV1MultiLocation,u64)',
            ResponseReady: '(u64,XcmV2Response)',
            Notified: '(u64,u8,u8)',
            NotifyOverweight: '(u64,u8,u8,u64,u64)',
            NotifyDispatchError: '(u64,u8,u8)',
            NotifyDecodeFailed: '(u64,u8,u8)',
            InvalidResponder: '(XcmV1MultiLocation,u64,Option<XcmV1MultiLocation>)',
            InvalidResponderVersion: '(XcmV1MultiLocation,u64)',
            ResponseTaken: 'u64',
            AssetsTrapped: '(H256,XcmV1MultiLocation,XcmVersionedMultiAssets)',
            VersionChangeNotified: '(XcmV1MultiLocation,u32)',
            SupportedVersionChanged: '(XcmV1MultiLocation,u32)',
            NotifyTargetSendFail: '(XcmV1MultiLocation,u64,XcmV2TraitsError)',
            NotifyTargetMigrationFail: '(XcmVersionedMultiLocation,u64)'
        }
    },
    /**
     * Lookup75: xcm::v2::traits::Outcome
     **/
    XcmV2TraitsOutcome: {
        _enum: {
            Complete: 'u64',
            Incomplete: '(u64,XcmV2TraitsError)',
            Error: 'XcmV2TraitsError'
        }
    },
    /**
     * Lookup76: xcm::v2::Xcm<Call>
     **/
    XcmV2Xcm: 'Vec<XcmV2Instruction>',
    /**
     * Lookup78: xcm::v2::Instruction<Call>
     **/
    XcmV2Instruction: {
        _enum: {
            WithdrawAsset: 'XcmV1MultiassetMultiAssets',
            ReserveAssetDeposited: 'XcmV1MultiassetMultiAssets',
            ReceiveTeleportedAsset: 'XcmV1MultiassetMultiAssets',
            QueryResponse: {
                queryId: 'Compact<u64>',
                response: 'XcmV2Response',
                maxWeight: 'Compact<u64>',
            },
            TransferAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                beneficiary: 'XcmV1MultiLocation',
            },
            TransferReserveAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                dest: 'XcmV1MultiLocation',
                xcm: 'XcmV2Xcm',
            },
            Transact: {
                originType: 'XcmV0OriginKind',
                requireWeightAtMost: 'Compact<u64>',
                call: 'XcmDoubleEncoded',
            },
            HrmpNewChannelOpenRequest: {
                sender: 'Compact<u32>',
                maxMessageSize: 'Compact<u32>',
                maxCapacity: 'Compact<u32>',
            },
            HrmpChannelAccepted: {
                recipient: 'Compact<u32>',
            },
            HrmpChannelClosing: {
                initiator: 'Compact<u32>',
                sender: 'Compact<u32>',
                recipient: 'Compact<u32>',
            },
            ClearOrigin: 'Null',
            DescendOrigin: 'XcmV1MultilocationJunctions',
            ReportError: {
                queryId: 'Compact<u64>',
                dest: 'XcmV1MultiLocation',
                maxResponseWeight: 'Compact<u64>',
            },
            DepositAsset: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                maxAssets: 'Compact<u32>',
                beneficiary: 'XcmV1MultiLocation',
            },
            DepositReserveAsset: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                maxAssets: 'Compact<u32>',
                dest: 'XcmV1MultiLocation',
                xcm: 'XcmV2Xcm',
            },
            ExchangeAsset: {
                give: 'XcmV1MultiassetMultiAssetFilter',
                receive: 'XcmV1MultiassetMultiAssets',
            },
            InitiateReserveWithdraw: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                reserve: 'XcmV1MultiLocation',
                xcm: 'XcmV2Xcm',
            },
            InitiateTeleport: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                dest: 'XcmV1MultiLocation',
                xcm: 'XcmV2Xcm',
            },
            QueryHolding: {
                queryId: 'Compact<u64>',
                dest: 'XcmV1MultiLocation',
                assets: 'XcmV1MultiassetMultiAssetFilter',
                maxResponseWeight: 'Compact<u64>',
            },
            BuyExecution: {
                fees: 'XcmV1MultiAsset',
                weightLimit: 'XcmV2WeightLimit',
            },
            RefundSurplus: 'Null',
            SetErrorHandler: 'XcmV2Xcm',
            SetAppendix: 'XcmV2Xcm',
            ClearError: 'Null',
            ClaimAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                ticket: 'XcmV1MultiLocation',
            },
            Trap: 'Compact<u64>',
            SubscribeVersion: {
                queryId: 'Compact<u64>',
                maxResponseWeight: 'Compact<u64>',
            },
            UnsubscribeVersion: 'Null'
        }
    },
    /**
     * Lookup79: xcm::v1::multiasset::MultiAssets
     **/
    XcmV1MultiassetMultiAssets: 'Vec<XcmV1MultiAsset>',
    /**
     * Lookup81: xcm::v1::multiasset::MultiAsset
     **/
    XcmV1MultiAsset: {
        id: 'XcmV1MultiassetAssetId',
        fun: 'XcmV1MultiassetFungibility'
    },
    /**
     * Lookup82: xcm::v1::multiasset::AssetId
     **/
    XcmV1MultiassetAssetId: {
        _enum: {
            Concrete: 'XcmV1MultiLocation',
            Abstract: 'Bytes'
        }
    },
    /**
     * Lookup83: xcm::v1::multiasset::Fungibility
     **/
    XcmV1MultiassetFungibility: {
        _enum: {
            Fungible: 'Compact<u128>',
            NonFungible: 'XcmV1MultiassetAssetInstance'
        }
    },
    /**
     * Lookup84: xcm::v1::multiasset::AssetInstance
     **/
    XcmV1MultiassetAssetInstance: {
        _enum: {
            Undefined: 'Null',
            Index: 'Compact<u128>',
            Array4: '[u8;4]',
            Array8: '[u8;8]',
            Array16: '[u8;16]',
            Array32: '[u8;32]',
            Blob: 'Bytes'
        }
    },
    /**
     * Lookup86: xcm::v2::Response
     **/
    XcmV2Response: {
        _enum: {
            Null: 'Null',
            Assets: 'XcmV1MultiassetMultiAssets',
            ExecutionResult: 'Option<(u32,XcmV2TraitsError)>',
            Version: 'u32'
        }
    },
    /**
     * Lookup89: xcm::v0::OriginKind
     **/
    XcmV0OriginKind: {
        _enum: ['Native', 'SovereignAccount', 'Superuser', 'Xcm']
    },
    /**
     * Lookup90: xcm::double_encoded::DoubleEncoded<T>
     **/
    XcmDoubleEncoded: {
        encoded: 'Bytes'
    },
    /**
     * Lookup91: xcm::v1::multiasset::MultiAssetFilter
     **/
    XcmV1MultiassetMultiAssetFilter: {
        _enum: {
            Definite: 'XcmV1MultiassetMultiAssets',
            Wild: 'XcmV1MultiassetWildMultiAsset'
        }
    },
    /**
     * Lookup92: xcm::v1::multiasset::WildMultiAsset
     **/
    XcmV1MultiassetWildMultiAsset: {
        _enum: {
            All: 'Null',
            AllOf: {
                id: 'XcmV1MultiassetAssetId',
                fun: 'XcmV1MultiassetWildFungibility'
            }
        }
    },
    /**
     * Lookup93: xcm::v1::multiasset::WildFungibility
     **/
    XcmV1MultiassetWildFungibility: {
        _enum: ['Fungible', 'NonFungible']
    },
    /**
     * Lookup94: xcm::v2::WeightLimit
     **/
    XcmV2WeightLimit: {
        _enum: {
            Unlimited: 'Null',
            Limited: 'Compact<u64>'
        }
    },
    /**
     * Lookup96: xcm::VersionedMultiAssets
     **/
    XcmVersionedMultiAssets: {
        _enum: {
            V0: 'Vec<XcmV0MultiAsset>',
            V1: 'XcmV1MultiassetMultiAssets'
        }
    },
    /**
     * Lookup98: xcm::v0::multi_asset::MultiAsset
     **/
    XcmV0MultiAsset: {
        _enum: {
            None: 'Null',
            All: 'Null',
            AllFungible: 'Null',
            AllNonFungible: 'Null',
            AllAbstractFungible: {
                id: 'Bytes',
            },
            AllAbstractNonFungible: {
                class: 'Bytes',
            },
            AllConcreteFungible: {
                id: 'XcmV0MultiLocation',
            },
            AllConcreteNonFungible: {
                class: 'XcmV0MultiLocation',
            },
            AbstractFungible: {
                id: 'Bytes',
                amount: 'Compact<u128>',
            },
            AbstractNonFungible: {
                class: 'Bytes',
                instance: 'XcmV1MultiassetAssetInstance',
            },
            ConcreteFungible: {
                id: 'XcmV0MultiLocation',
                amount: 'Compact<u128>',
            },
            ConcreteNonFungible: {
                class: 'XcmV0MultiLocation',
                instance: 'XcmV1MultiassetAssetInstance'
            }
        }
    },
    /**
     * Lookup99: cumulus_pallet_xcm::pallet::Event<T>
     **/
    CumulusPalletXcmEvent: {
        _enum: {
            InvalidFormat: '[u8;8]',
            UnsupportedVersion: '[u8;8]',
            ExecutedDownward: '([u8;8],XcmV2TraitsOutcome)'
        }
    },
    /**
     * Lookup100: cumulus_pallet_dmp_queue::pallet::Event<T>
     **/
    CumulusPalletDmpQueueEvent: {
        _enum: {
            InvalidFormat: {
                messageId: '[u8;32]',
            },
            UnsupportedVersion: {
                messageId: '[u8;32]',
            },
            ExecutedDownward: {
                messageId: '[u8;32]',
                outcome: 'XcmV2TraitsOutcome',
            },
            WeightExhausted: {
                messageId: '[u8;32]',
                remainingWeight: 'u64',
                requiredWeight: 'u64',
            },
            OverweightEnqueued: {
                messageId: '[u8;32]',
                overweightIndex: 'u64',
                requiredWeight: 'u64',
            },
            OverweightServiced: {
                overweightIndex: 'u64',
                weightUsed: 'u64'
            }
        }
    },
    /**
     * Lookup101: orml_xtokens::module::Event<T>
     **/
    OrmlXtokensModuleEvent: {
        _enum: {
            TransferredMultiAssets: {
                sender: 'AccountId32',
                assets: 'XcmV1MultiassetMultiAssets',
                fee: 'XcmV1MultiAsset',
                dest: 'XcmV1MultiLocation'
            }
        }
    },
    /**
     * Lookup102: orml_unknown_tokens::module::Event
     **/
    OrmlUnknownTokensModuleEvent: {
        _enum: {
            Deposited: {
                asset: 'XcmV1MultiAsset',
                who: 'XcmV1MultiLocation',
            },
            Withdrawn: {
                asset: 'XcmV1MultiAsset',
                who: 'XcmV1MultiLocation'
            }
        }
    },
    /**
     * Lookup103: pallet_sudo::pallet::Event<T>
     **/
    PalletSudoEvent: {
        _enum: {
            Sudid: {
                sudoResult: 'Result<Null, SpRuntimeDispatchError>',
            },
            KeyChanged: {
                oldSudoer: 'Option<AccountId32>',
            },
            SudoAsDone: {
                sudoResult: 'Result<Null, SpRuntimeDispatchError>'
            }
        }
    },
    /**
     * Lookup105: pallet_treasury::pallet::Event<T, I>
     **/
    PalletTreasuryEvent: {
        _enum: {
            Proposed: {
                proposalIndex: 'u32',
            },
            Spending: {
                budgetRemaining: 'u128',
            },
            Awarded: {
                proposalIndex: 'u32',
                award: 'u128',
                account: 'AccountId32',
            },
            Rejected: {
                proposalIndex: 'u32',
                slashed: 'u128',
            },
            Burnt: {
                burntFunds: 'u128',
            },
            Rollover: {
                rolloverBalance: 'u128',
            },
            Deposit: {
                value: 'u128',
            },
            SpendApproved: {
                proposalIndex: 'u32',
                amount: 'u128',
                beneficiary: 'AccountId32'
            }
        }
    },
    /**
     * Lookup106: pallet_collective::pallet::Event<T, I>
     **/
    PalletCollectiveEvent: {
        _enum: {
            Proposed: {
                account: 'AccountId32',
                proposalIndex: 'u32',
                proposalHash: 'H256',
                threshold: 'u32',
            },
            Voted: {
                account: 'AccountId32',
                proposalHash: 'H256',
                voted: 'bool',
                yes: 'u32',
                no: 'u32',
            },
            Approved: {
                proposalHash: 'H256',
            },
            Disapproved: {
                proposalHash: 'H256',
            },
            Executed: {
                proposalHash: 'H256',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            MemberExecuted: {
                proposalHash: 'H256',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            Closed: {
                proposalHash: 'H256',
                yes: 'u32',
                no: 'u32'
            }
        }
    },
    /**
     * Lookup108: pallet_membership::pallet::Event<T, I>
     **/
    PalletMembershipEvent: {
        _enum: ['MemberAdded', 'MemberRemoved', 'MembersSwapped', 'MembersReset', 'KeyChanged', 'Dummy']
    },
    /**
     * Lookup109: pallet_bounties::pallet::Event<T, I>
     **/
    PalletBountiesEvent: {
        _enum: {
            BountyProposed: {
                index: 'u32',
            },
            BountyRejected: {
                index: 'u32',
                bond: 'u128',
            },
            BountyBecameActive: {
                index: 'u32',
            },
            BountyAwarded: {
                index: 'u32',
                beneficiary: 'AccountId32',
            },
            BountyClaimed: {
                index: 'u32',
                payout: 'u128',
                beneficiary: 'AccountId32',
            },
            BountyCanceled: {
                index: 'u32',
            },
            BountyExtended: {
                index: 'u32'
            }
        }
    },
    /**
     * Lookup110: pallet_preimage::pallet::Event<T>
     **/
    PalletPreimageEvent: {
        _enum: {
            Noted: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256',
            },
            Requested: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256',
            },
            Cleared: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256'
            }
        }
    },
    /**
     * Lookup111: pallet_scheduler::pallet::Event<T>
     **/
    PalletSchedulerEvent: {
        _enum: {
            Scheduled: {
                when: 'u32',
                index: 'u32',
            },
            Canceled: {
                when: 'u32',
                index: 'u32',
            },
            Dispatched: {
                task: '(u32,u32)',
                id: 'Option<Bytes>',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            CallLookupFailed: {
                task: '(u32,u32)',
                id: 'Option<Bytes>',
                error: 'FrameSupportScheduleLookupError'
            }
        }
    },
    /**
     * Lookup114: frame_support::traits::schedule::LookupError
     **/
    FrameSupportScheduleLookupError: {
        _enum: ['Unknown', 'BadFormat']
    },
    /**
     * Lookup115: pallet_democracy::pallet::Event<T>
     **/
    PalletDemocracyEvent: {
        _enum: {
            Proposed: {
                proposalIndex: 'u32',
                deposit: 'u128',
            },
            Tabled: {
                proposalIndex: 'u32',
                deposit: 'u128',
                depositors: 'Vec<AccountId32>',
            },
            ExternalTabled: 'Null',
            Started: {
                refIndex: 'u32',
                threshold: 'PalletDemocracyVoteThreshold',
            },
            Passed: {
                refIndex: 'u32',
            },
            NotPassed: {
                refIndex: 'u32',
            },
            Cancelled: {
                refIndex: 'u32',
            },
            Executed: {
                refIndex: 'u32',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            Delegated: {
                who: 'AccountId32',
                target: 'AccountId32',
            },
            Undelegated: {
                account: 'AccountId32',
            },
            Vetoed: {
                who: 'AccountId32',
                proposalHash: 'H256',
                until: 'u32',
            },
            PreimageNoted: {
                proposalHash: 'H256',
                who: 'AccountId32',
                deposit: 'u128',
            },
            PreimageUsed: {
                proposalHash: 'H256',
                provider: 'AccountId32',
                deposit: 'u128',
            },
            PreimageInvalid: {
                proposalHash: 'H256',
                refIndex: 'u32',
            },
            PreimageMissing: {
                proposalHash: 'H256',
                refIndex: 'u32',
            },
            PreimageReaped: {
                proposalHash: 'H256',
                provider: 'AccountId32',
                deposit: 'u128',
                reaper: 'AccountId32',
            },
            Blacklisted: {
                proposalHash: 'H256',
            },
            Voted: {
                voter: 'AccountId32',
                refIndex: 'u32',
                vote: 'PalletDemocracyVoteAccountVote',
            },
            Seconded: {
                seconder: 'AccountId32',
                propIndex: 'u32',
            },
            ProposalCanceled: {
                propIndex: 'u32'
            }
        }
    },
    /**
     * Lookup117: pallet_democracy::vote_threshold::VoteThreshold
     **/
    PalletDemocracyVoteThreshold: {
        _enum: ['SuperMajorityApprove', 'SuperMajorityAgainst', 'SimpleMajority']
    },
    /**
     * Lookup118: pallet_democracy::vote::AccountVote<Balance>
     **/
    PalletDemocracyVoteAccountVote: {
        _enum: {
            Standard: {
                vote: 'Vote',
                balance: 'u128',
            },
            Split: {
                aye: 'u128',
                nay: 'u128'
            }
        }
    },
    /**
     * Lookup120: pallet_multisig::pallet::Event<T>
     **/
    PalletMultisigEvent: {
        _enum: {
            NewMultisig: {
                approving: 'AccountId32',
                multisig: 'AccountId32',
                callHash: '[u8;32]',
            },
            MultisigApproval: {
                approving: 'AccountId32',
                timepoint: 'PalletMultisigTimepoint',
                multisig: 'AccountId32',
                callHash: '[u8;32]',
            },
            MultisigExecuted: {
                approving: 'AccountId32',
                timepoint: 'PalletMultisigTimepoint',
                multisig: 'AccountId32',
                callHash: '[u8;32]',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            MultisigCancelled: {
                cancelling: 'AccountId32',
                timepoint: 'PalletMultisigTimepoint',
                multisig: 'AccountId32',
                callHash: '[u8;32]'
            }
        }
    },
    /**
     * Lookup121: pallet_multisig::Timepoint<BlockNumber>
     **/
    PalletMultisigTimepoint: {
        height: 'u32',
        index: 'u32'
    },
    /**
     * Lookup122: pallet_automation_time::pallet::Event<T>
     **/
    PalletAutomationTimeEvent: {
        _enum: {
            TaskScheduled: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            TaskCancelled: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            Notify: {
                message: 'Bytes',
            },
            TaskNotFound: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            SuccessfullyTransferredFunds: {
                taskId: 'H256',
            },
            XcmpTaskSucceeded: {
                taskId: 'H256',
                paraId: 'u32',
            },
            XcmpTaskFailed: {
                taskId: 'H256',
                paraId: 'u32',
                error: 'SpRuntimeDispatchError',
            },
            TransferFailed: {
                taskId: 'H256',
                error: 'SpRuntimeDispatchError',
            },
            SuccesfullyAutoCompoundedDelegatorStake: {
                taskId: 'H256',
                amount: 'u128',
            },
            AutoCompoundDelegatorStakeFailed: {
                taskId: 'H256',
                errorMessage: 'Bytes',
                error: 'SpRuntimeDispatchErrorWithPostInfo',
            },
            TaskMissed: {
                who: 'AccountId32',
                taskId: 'H256',
                executionTime: 'u64',
            },
            DynamicDispatchResult: {
                who: 'AccountId32',
                taskId: 'H256',
                result: 'Result<Null, SpRuntimeDispatchError>',
            },
            CallCannotBeDecoded: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            TaskRescheduled: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            TaskNotRescheduled: {
                who: 'AccountId32',
                taskId: 'H256',
                error: 'SpRuntimeDispatchError',
            },
            TaskFailedToReschedule: {
                who: 'AccountId32',
                taskId: 'H256',
                error: 'SpRuntimeDispatchError'
            }
        }
    },
    /**
     * Lookup123: sp_runtime::DispatchErrorWithPostInfo<frame_support::weights::PostDispatchInfo>
     **/
    SpRuntimeDispatchErrorWithPostInfo: {
        postInfo: 'FrameSupportWeightsPostDispatchInfo',
        error: 'SpRuntimeDispatchError'
    },
    /**
     * Lookup124: frame_support::weights::PostDispatchInfo
     **/
    FrameSupportWeightsPostDispatchInfo: {
        actualWeight: 'Option<u64>',
        paysFee: 'FrameSupportWeightsPays'
    },
    /**
     * Lookup126: pallet_vesting::pallet::Event<T>
     **/
    PalletVestingEvent: {
        _enum: {
            Vested: {
                account: 'AccountId32',
                amount: 'u128',
            },
            VestFailed: {
                account: 'AccountId32',
                amount: 'u128',
                error: 'SpRuntimeDispatchError'
            }
        }
    },
    /**
     * Lookup127: pallet_xcmp_handler::pallet::Event<T>
     **/
    PalletXcmpHandlerEvent: {
        _enum: {
            XcmDataAdded: {
                paraId: 'u32',
                currencyId: 'u32',
            },
            XcmDataRemoved: {
                paraId: 'u32',
                currencyId: 'u32',
            },
            XcmSent: {
                paraId: 'u32',
            },
            XcmTransactedLocally: 'Null',
            XcmFeesPaid: {
                source: 'AccountId32',
                dest: 'AccountId32',
            },
            XcmFeesFailed: {
                source: 'AccountId32',
                dest: 'AccountId32',
                error: 'SpRuntimeDispatchError'
            }
        }
    },
    /**
     * Lookup128: pallet_automation_price::pallet::Event<T>
     **/
    PalletAutomationPriceEvent: {
        _enum: {
            TaskScheduled: {
                who: 'AccountId32',
                taskId: 'H256',
            },
            Notify: {
                message: 'Bytes',
            },
            TaskNotFound: {
                taskId: 'H256',
            },
            AssetCreated: {
                asset: 'Bytes',
            },
            AssetUpdated: {
                asset: 'Bytes',
            },
            AssetDeleted: {
                asset: 'Bytes',
            },
            AssetPeriodReset: {
                asset: 'Bytes',
            },
            SuccessfullyTransferredFunds: {
                taskId: 'H256',
            },
            TransferFailed: {
                taskId: 'H256',
                error: 'SpRuntimeDispatchError'
            }
        }
    },
    /**
     * Lookup129: frame_system::Phase
     **/
    FrameSystemPhase: {
        _enum: {
            ApplyExtrinsic: 'u32',
            Finalization: 'Null',
            Initialization: 'Null'
        }
    },
    /**
     * Lookup132: frame_system::LastRuntimeUpgradeInfo
     **/
    FrameSystemLastRuntimeUpgradeInfo: {
        specVersion: 'Compact<u32>',
        specName: 'Text'
    },
    /**
     * Lookup134: frame_system::pallet::Call<T>
     **/
    FrameSystemCall: {
        _enum: {
            fill_block: {
                ratio: 'Perbill',
            },
            remark: {
                remark: 'Bytes',
            },
            set_heap_pages: {
                pages: 'u64',
            },
            set_code: {
                code: 'Bytes',
            },
            set_code_without_checks: {
                code: 'Bytes',
            },
            set_storage: {
                items: 'Vec<(Bytes,Bytes)>',
            },
            kill_storage: {
                _alias: {
                    keys_: 'keys',
                },
                keys_: 'Vec<Bytes>',
            },
            kill_prefix: {
                prefix: 'Bytes',
                subkeys: 'u32',
            },
            remark_with_event: {
                remark: 'Bytes'
            }
        }
    },
    /**
     * Lookup138: frame_system::limits::BlockWeights
     **/
    FrameSystemLimitsBlockWeights: {
        baseBlock: 'u64',
        maxBlock: 'u64',
        perClass: 'FrameSupportWeightsPerDispatchClassWeightsPerClass'
    },
    /**
     * Lookup139: frame_support::weights::PerDispatchClass<frame_system::limits::WeightsPerClass>
     **/
    FrameSupportWeightsPerDispatchClassWeightsPerClass: {
        normal: 'FrameSystemLimitsWeightsPerClass',
        operational: 'FrameSystemLimitsWeightsPerClass',
        mandatory: 'FrameSystemLimitsWeightsPerClass'
    },
    /**
     * Lookup140: frame_system::limits::WeightsPerClass
     **/
    FrameSystemLimitsWeightsPerClass: {
        baseExtrinsic: 'u64',
        maxExtrinsic: 'Option<u64>',
        maxTotal: 'Option<u64>',
        reserved: 'Option<u64>'
    },
    /**
     * Lookup141: frame_system::limits::BlockLength
     **/
    FrameSystemLimitsBlockLength: {
        max: 'FrameSupportWeightsPerDispatchClassU32'
    },
    /**
     * Lookup142: frame_support::weights::PerDispatchClass<T>
     **/
    FrameSupportWeightsPerDispatchClassU32: {
        normal: 'u32',
        operational: 'u32',
        mandatory: 'u32'
    },
    /**
     * Lookup143: frame_support::weights::RuntimeDbWeight
     **/
    FrameSupportWeightsRuntimeDbWeight: {
        read: 'u64',
        write: 'u64'
    },
    /**
     * Lookup144: sp_version::RuntimeVersion
     **/
    SpVersionRuntimeVersion: {
        specName: 'Text',
        implName: 'Text',
        authoringVersion: 'u32',
        specVersion: 'u32',
        implVersion: 'u32',
        apis: 'Vec<([u8;8],u32)>',
        transactionVersion: 'u32',
        stateVersion: 'u8'
    },
    /**
     * Lookup148: frame_system::pallet::Error<T>
     **/
    FrameSystemError: {
        _enum: ['InvalidSpecName', 'SpecVersionNeedsToIncrease', 'FailedToExtractRuntimeVersion', 'NonDefaultComposite', 'NonZeroRefCount', 'CallFiltered']
    },
    /**
     * Lookup149: polkadot_primitives::v2::PersistedValidationData<primitive_types::H256, N>
     **/
    PolkadotPrimitivesV2PersistedValidationData: {
        parentHead: 'Bytes',
        relayParentNumber: 'u32',
        relayParentStorageRoot: 'H256',
        maxPovSize: 'u32'
    },
    /**
     * Lookup152: polkadot_primitives::v2::UpgradeRestriction
     **/
    PolkadotPrimitivesV2UpgradeRestriction: {
        _enum: ['Present']
    },
    /**
     * Lookup153: sp_trie::storage_proof::StorageProof
     **/
    SpTrieStorageProof: {
        trieNodes: 'BTreeSet<Bytes>'
    },
    /**
     * Lookup155: cumulus_pallet_parachain_system::relay_state_snapshot::MessagingStateSnapshot
     **/
    CumulusPalletParachainSystemRelayStateSnapshotMessagingStateSnapshot: {
        dmqMqcHead: 'H256',
        relayDispatchQueueSize: '(u32,u32)',
        ingressChannels: 'Vec<(u32,PolkadotPrimitivesV2AbridgedHrmpChannel)>',
        egressChannels: 'Vec<(u32,PolkadotPrimitivesV2AbridgedHrmpChannel)>'
    },
    /**
     * Lookup158: polkadot_primitives::v2::AbridgedHrmpChannel
     **/
    PolkadotPrimitivesV2AbridgedHrmpChannel: {
        maxCapacity: 'u32',
        maxTotalSize: 'u32',
        maxMessageSize: 'u32',
        msgCount: 'u32',
        totalSize: 'u32',
        mqcHead: 'Option<H256>'
    },
    /**
     * Lookup159: polkadot_primitives::v2::AbridgedHostConfiguration
     **/
    PolkadotPrimitivesV2AbridgedHostConfiguration: {
        maxCodeSize: 'u32',
        maxHeadDataSize: 'u32',
        maxUpwardQueueCount: 'u32',
        maxUpwardQueueSize: 'u32',
        maxUpwardMessageSize: 'u32',
        maxUpwardMessageNumPerCandidate: 'u32',
        hrmpMaxMessageNumPerCandidate: 'u32',
        validationUpgradeCooldown: 'u32',
        validationUpgradeDelay: 'u32'
    },
    /**
     * Lookup165: polkadot_core_primitives::OutboundHrmpMessage<polkadot_parachain::primitives::Id>
     **/
    PolkadotCorePrimitivesOutboundHrmpMessage: {
        recipient: 'u32',
        data: 'Bytes'
    },
    /**
     * Lookup166: cumulus_pallet_parachain_system::pallet::Call<T>
     **/
    CumulusPalletParachainSystemCall: {
        _enum: {
            set_validation_data: {
                data: 'CumulusPrimitivesParachainInherentParachainInherentData',
            },
            sudo_send_upward_message: {
                message: 'Bytes',
            },
            authorize_upgrade: {
                codeHash: 'H256',
            },
            enact_authorized_upgrade: {
                code: 'Bytes'
            }
        }
    },
    /**
     * Lookup167: cumulus_primitives_parachain_inherent::ParachainInherentData
     **/
    CumulusPrimitivesParachainInherentParachainInherentData: {
        validationData: 'PolkadotPrimitivesV2PersistedValidationData',
        relayChainState: 'SpTrieStorageProof',
        downwardMessages: 'Vec<PolkadotCorePrimitivesInboundDownwardMessage>',
        horizontalMessages: 'BTreeMap<u32, Vec<PolkadotCorePrimitivesInboundHrmpMessage>>'
    },
    /**
     * Lookup169: polkadot_core_primitives::InboundDownwardMessage<BlockNumber>
     **/
    PolkadotCorePrimitivesInboundDownwardMessage: {
        sentAt: 'u32',
        msg: 'Bytes'
    },
    /**
     * Lookup172: polkadot_core_primitives::InboundHrmpMessage<BlockNumber>
     **/
    PolkadotCorePrimitivesInboundHrmpMessage: {
        sentAt: 'u32',
        data: 'Bytes'
    },
    /**
     * Lookup175: cumulus_pallet_parachain_system::pallet::Error<T>
     **/
    CumulusPalletParachainSystemError: {
        _enum: ['OverlappingUpgrades', 'ProhibitedByPolkadot', 'TooBig', 'ValidationDataNotAvailable', 'HostConfigurationNotAvailable', 'NotScheduled', 'NothingAuthorized', 'Unauthorized']
    },
    /**
     * Lookup176: pallet_timestamp::pallet::Call<T>
     **/
    PalletTimestampCall: {
        _enum: {
            set: {
                now: 'Compact<u64>'
            }
        }
    },
    /**
     * Lookup178: pallet_balances::BalanceLock<Balance>
     **/
    PalletBalancesBalanceLock: {
        id: '[u8;8]',
        amount: 'u128',
        reasons: 'PalletBalancesReasons'
    },
    /**
     * Lookup179: pallet_balances::Reasons
     **/
    PalletBalancesReasons: {
        _enum: ['Fee', 'Misc', 'All']
    },
    /**
     * Lookup182: pallet_balances::ReserveData<ReserveIdentifier, Balance>
     **/
    PalletBalancesReserveData: {
        id: '[u8;8]',
        amount: 'u128'
    },
    /**
     * Lookup184: pallet_balances::Releases
     **/
    PalletBalancesReleases: {
        _enum: ['V1_0_0', 'V2_0_0']
    },
    /**
     * Lookup185: pallet_balances::pallet::Call<T, I>
     **/
    PalletBalancesCall: {
        _enum: {
            transfer: {
                dest: 'MultiAddress',
                value: 'Compact<u128>',
            },
            set_balance: {
                who: 'MultiAddress',
                newFree: 'Compact<u128>',
                newReserved: 'Compact<u128>',
            },
            force_transfer: {
                source: 'MultiAddress',
                dest: 'MultiAddress',
                value: 'Compact<u128>',
            },
            transfer_keep_alive: {
                dest: 'MultiAddress',
                value: 'Compact<u128>',
            },
            transfer_all: {
                dest: 'MultiAddress',
                keepAlive: 'bool',
            },
            force_unreserve: {
                who: 'MultiAddress',
                amount: 'u128'
            }
        }
    },
    /**
     * Lookup188: pallet_balances::pallet::Error<T, I>
     **/
    PalletBalancesError: {
        _enum: ['VestingBalance', 'LiquidityRestrictions', 'InsufficientBalance', 'ExistentialDeposit', 'KeepAlive', 'ExistingVestingSchedule', 'DeadAccount', 'TooManyReserves']
    },
    /**
     * Lookup191: orml_tokens::BalanceLock<Balance>
     **/
    OrmlTokensBalanceLock: {
        id: '[u8;8]',
        amount: 'u128'
    },
    /**
     * Lookup193: orml_tokens::AccountData<Balance>
     **/
    OrmlTokensAccountData: {
        free: 'u128',
        reserved: 'u128',
        frozen: 'u128'
    },
    /**
     * Lookup195: orml_tokens::ReserveData<ReserveIdentifier, Balance>
     **/
    OrmlTokensReserveData: {
        id: '[u8;8]',
        amount: 'u128'
    },
    /**
     * Lookup197: orml_tokens::module::Error<T>
     **/
    OrmlTokensModuleError: {
        _enum: ['BalanceTooLow', 'AmountIntoBalanceFailed', 'LiquidityRestrictions', 'MaxLocksExceeded', 'KeepAlive', 'ExistentialDeposit', 'DeadAccount', 'TooManyReserves']
    },
    /**
     * Lookup198: orml_currencies::module::Call<T>
     **/
    OrmlCurrenciesModuleCall: {
        _enum: {
            transfer: {
                dest: 'MultiAddress',
                currencyId: 'u32',
                amount: 'Compact<u128>',
            },
            transfer_native_currency: {
                dest: 'MultiAddress',
                amount: 'Compact<u128>',
            },
            update_balance: {
                who: 'MultiAddress',
                currencyId: 'u32',
                amount: 'i128'
            }
        }
    },
    /**
     * Lookup200: orml_currencies::module::Error<T>
     **/
    OrmlCurrenciesModuleError: {
        _enum: ['AmountIntoBalanceFailed', 'BalanceTooLow', 'DepositFailed']
    },
    /**
     * Lookup202: pallet_transaction_payment::Releases
     **/
    PalletTransactionPaymentReleases: {
        _enum: ['V1Ancient', 'V2']
    },
    /**
     * Lookup203: orml_asset_registry::module::Call<T>
     **/
    OrmlAssetRegistryModuleCall: {
        _enum: {
            register_asset: {
                metadata: 'OrmlTraitsAssetRegistryAssetMetadata',
                assetId: 'Option<u32>',
            },
            update_asset: {
                assetId: 'u32',
                decimals: 'Option<u32>',
                name: 'Option<Bytes>',
                symbol: 'Option<Bytes>',
                existentialDeposit: 'Option<u128>',
                location: 'Option<Option<XcmVersionedMultiLocation>>',
                additional: 'Option<PrimitivesAssetsCustomMetadata>'
            }
        }
    },
    /**
     * Lookup207: orml_asset_registry::module::Error<T>
     **/
    OrmlAssetRegistryModuleError: {
        _enum: ['AssetNotFound', 'BadVersion', 'InvalidAssetId', 'ConflictingLocation', 'ConflictingAssetId']
    },
    /**
     * Lookup209: pallet_authorship::UncleEntryItem<BlockNumber, primitive_types::H256, sp_core::crypto::AccountId32>
     **/
    PalletAuthorshipUncleEntryItem: {
        _enum: {
            InclusionHeight: 'u32',
            Uncle: '(H256,Option<AccountId32>)'
        }
    },
    /**
     * Lookup211: pallet_authorship::pallet::Call<T>
     **/
    PalletAuthorshipCall: {
        _enum: {
            set_uncles: {
                newUncles: 'Vec<SpRuntimeHeader>'
            }
        }
    },
    /**
     * Lookup213: sp_runtime::generic::header::Header<Number, sp_runtime::traits::BlakeTwo256>
     **/
    SpRuntimeHeader: {
        parentHash: 'H256',
        number: 'Compact<u32>',
        stateRoot: 'H256',
        extrinsicsRoot: 'H256',
        digest: 'SpRuntimeDigest'
    },
    /**
     * Lookup214: sp_runtime::traits::BlakeTwo256
     **/
    SpRuntimeBlakeTwo256: 'Null',
    /**
     * Lookup215: pallet_authorship::pallet::Error<T>
     **/
    PalletAuthorshipError: {
        _enum: ['InvalidUncleParent', 'UnclesAlreadySet', 'TooManyUncles', 'GenesisUncle', 'TooHighUncle', 'UncleAlreadyIncluded', 'OldUncle']
    },
    /**
     * Lookup218: turing_runtime::SessionKeys
     **/
    TuringRuntimeSessionKeys: {
        aura: 'SpConsensusAuraSr25519AppSr25519Public'
    },
    /**
     * Lookup219: sp_consensus_aura::sr25519::app_sr25519::Public
     **/
    SpConsensusAuraSr25519AppSr25519Public: 'SpCoreSr25519Public',
    /**
     * Lookup220: sp_core::sr25519::Public
     **/
    SpCoreSr25519Public: '[u8;32]',
    /**
     * Lookup223: sp_core::crypto::KeyTypeId
     **/
    SpCoreCryptoKeyTypeId: '[u8;4]',
    /**
     * Lookup224: pallet_session::pallet::Call<T>
     **/
    PalletSessionCall: {
        _enum: {
            set_keys: {
                _alias: {
                    keys_: 'keys',
                },
                keys_: 'TuringRuntimeSessionKeys',
                proof: 'Bytes',
            },
            purge_keys: 'Null'
        }
    },
    /**
     * Lookup225: pallet_session::pallet::Error<T>
     **/
    PalletSessionError: {
        _enum: ['InvalidProof', 'NoAssociatedValidatorId', 'DuplicatedKey', 'NoKeys', 'NoAccount']
    },
    /**
     * Lookup226: pallet_parachain_staking::types::ParachainBondConfig<sp_core::crypto::AccountId32>
     **/
    PalletParachainStakingParachainBondConfig: {
        account: 'AccountId32',
        percent: 'Percent'
    },
    /**
     * Lookup227: pallet_parachain_staking::types::RoundInfo<BlockNumber>
     **/
    PalletParachainStakingRoundInfo: {
        current: 'u32',
        first: 'u32',
        length: 'u32'
    },
    /**
     * Lookup228: pallet_parachain_staking::types::Delegator<sp_core::crypto::AccountId32, Balance>
     **/
    PalletParachainStakingDelegator: {
        id: 'AccountId32',
        delegations: 'PalletParachainStakingSetOrderedSet',
        total: 'u128',
        lessTotal: 'u128',
        status: 'PalletParachainStakingDelegatorStatus'
    },
    /**
     * Lookup229: pallet_parachain_staking::set::OrderedSet<pallet_parachain_staking::types::Bond<sp_core::crypto::AccountId32, Balance>>
     **/
    PalletParachainStakingSetOrderedSet: 'Vec<PalletParachainStakingBond>',
    /**
     * Lookup230: pallet_parachain_staking::types::Bond<sp_core::crypto::AccountId32, Balance>
     **/
    PalletParachainStakingBond: {
        owner: 'AccountId32',
        amount: 'u128'
    },
    /**
     * Lookup232: pallet_parachain_staking::types::DelegatorStatus
     **/
    PalletParachainStakingDelegatorStatus: {
        _enum: {
            Active: 'Null',
            Leaving: 'u32'
        }
    },
    /**
     * Lookup233: pallet_parachain_staking::types::CandidateMetadata<Balance>
     **/
    PalletParachainStakingCandidateMetadata: {
        bond: 'u128',
        delegationCount: 'u32',
        totalCounted: 'u128',
        lowestTopDelegationAmount: 'u128',
        highestBottomDelegationAmount: 'u128',
        lowestBottomDelegationAmount: 'u128',
        topCapacity: 'PalletParachainStakingCapacityStatus',
        bottomCapacity: 'PalletParachainStakingCapacityStatus',
        request: 'Option<PalletParachainStakingCandidateBondLessRequest>',
        status: 'PalletParachainStakingCollatorStatus'
    },
    /**
     * Lookup234: pallet_parachain_staking::types::CapacityStatus
     **/
    PalletParachainStakingCapacityStatus: {
        _enum: ['Full', 'Empty', 'Partial']
    },
    /**
     * Lookup236: pallet_parachain_staking::types::CandidateBondLessRequest<Balance>
     **/
    PalletParachainStakingCandidateBondLessRequest: {
        amount: 'u128',
        whenExecutable: 'u32'
    },
    /**
     * Lookup237: pallet_parachain_staking::types::CollatorStatus
     **/
    PalletParachainStakingCollatorStatus: {
        _enum: {
            Active: 'Null',
            Idle: 'Null',
            Leaving: 'u32'
        }
    },
    /**
     * Lookup239: pallet_parachain_staking::delegation_requests::ScheduledRequest<sp_core::crypto::AccountId32, Balance>
     **/
    PalletParachainStakingDelegationRequestsScheduledRequest: {
        delegator: 'AccountId32',
        whenExecutable: 'u32',
        action: 'PalletParachainStakingDelegationRequestsDelegationAction'
    },
    /**
     * Lookup240: pallet_parachain_staking::types::Delegations<sp_core::crypto::AccountId32, Balance>
     **/
    PalletParachainStakingDelegations: {
        delegations: 'Vec<PalletParachainStakingBond>',
        total: 'u128'
    },
    /**
     * Lookup242: pallet_parachain_staking::types::CollatorSnapshot<sp_core::crypto::AccountId32, Balance>
     **/
    PalletParachainStakingCollatorSnapshot: {
        bond: 'u128',
        delegations: 'Vec<PalletParachainStakingBond>',
        total: 'u128'
    },
    /**
     * Lookup243: pallet_parachain_staking::types::DelayedPayout<Balance>
     **/
    PalletParachainStakingDelayedPayout: {
        roundIssuance: 'u128',
        totalStakingReward: 'u128',
        collatorCommission: 'Perbill'
    },
    /**
     * Lookup244: pallet_parachain_staking::inflation::InflationInfo<Balance>
     **/
    PalletParachainStakingInflationInflationInfo: {
        expect: {
            min: 'u128',
            ideal: 'u128',
            max: 'u128'
        },
        annual: {
            min: 'Perbill',
            ideal: 'Perbill',
            max: 'Perbill'
        },
        round: {
            min: 'Perbill',
            ideal: 'Perbill',
            max: 'Perbill'
        }
    },
    /**
     * Lookup247: pallet_parachain_staking::pallet::Call<T>
     **/
    PalletParachainStakingCall: {
        _enum: {
            set_staking_expectations: {
                expectations: {
                    min: 'u128',
                    ideal: 'u128',
                    max: 'u128'
                },
            },
            set_inflation: {
                schedule: {
                    min: 'Perbill',
                    ideal: 'Perbill',
                    max: 'Perbill'
                },
            },
            set_parachain_bond_account: {
                _alias: {
                    new_: 'new',
                },
                new_: 'AccountId32',
            },
            set_parachain_bond_reserve_percent: {
                _alias: {
                    new_: 'new',
                },
                new_: 'Percent',
            },
            set_total_selected: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u32',
            },
            set_collator_commission: {
                _alias: {
                    new_: 'new',
                },
                new_: 'Perbill',
            },
            set_blocks_per_round: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u32',
            },
            join_candidates: {
                bond: 'u128',
                candidateCount: 'u32',
            },
            schedule_leave_candidates: {
                candidateCount: 'u32',
            },
            execute_leave_candidates: {
                candidate: 'AccountId32',
                candidateDelegationCount: 'u32',
            },
            cancel_leave_candidates: {
                candidateCount: 'u32',
            },
            go_offline: 'Null',
            go_online: 'Null',
            candidate_bond_more: {
                more: 'u128',
            },
            schedule_candidate_bond_less: {
                less: 'u128',
            },
            execute_candidate_bond_less: {
                candidate: 'AccountId32',
            },
            cancel_candidate_bond_less: 'Null',
            delegate: {
                candidate: 'AccountId32',
                amount: 'u128',
                candidateDelegationCount: 'u32',
                delegationCount: 'u32',
            },
            schedule_leave_delegators: 'Null',
            execute_leave_delegators: {
                delegator: 'AccountId32',
                delegationCount: 'u32',
            },
            cancel_leave_delegators: 'Null',
            schedule_revoke_delegation: {
                collator: 'AccountId32',
            },
            delegator_bond_more: {
                candidate: 'AccountId32',
                more: 'u128',
            },
            schedule_delegator_bond_less: {
                candidate: 'AccountId32',
                less: 'u128',
            },
            execute_delegation_request: {
                delegator: 'AccountId32',
                candidate: 'AccountId32',
            },
            cancel_delegation_request: {
                candidate: 'AccountId32',
            },
            hotfix_remove_delegation_requests_exited_candidates: {
                candidates: 'Vec<AccountId32>',
            },
            hotfix_migrate_delegators_from_reserve_to_locks: {
                delegators: 'Vec<AccountId32>',
            },
            hotfix_migrate_collators_from_reserve_to_locks: {
                collators: 'Vec<AccountId32>'
            }
        }
    },
    /**
     * Lookup248: pallet_parachain_staking::pallet::Error<T>
     **/
    PalletParachainStakingError: {
        _enum: ['DelegatorDNE', 'DelegatorDNEinTopNorBottom', 'DelegatorDNEInDelegatorSet', 'CandidateDNE', 'DelegationDNE', 'DelegatorExists', 'CandidateExists', 'CandidateBondBelowMin', 'InsufficientBalance', 'DelegatorBondBelowMin', 'DelegationBelowMin', 'AlreadyOffline', 'AlreadyActive', 'DelegatorAlreadyLeaving', 'DelegatorNotLeaving', 'DelegatorCannotLeaveYet', 'CannotDelegateIfLeaving', 'CandidateAlreadyLeaving', 'CandidateNotLeaving', 'CandidateCannotLeaveYet', 'CannotGoOnlineIfLeaving', 'ExceedMaxDelegationsPerDelegator', 'AlreadyDelegatedCandidate', 'InvalidSchedule', 'CannotSetBelowMin', 'RoundLengthMustBeAtLeastTotalSelectedCollators', 'NoWritingSameValue', 'TooLowCandidateCountWeightHintJoinCandidates', 'TooLowCandidateCountWeightHintCancelLeaveCandidates', 'TooLowCandidateCountToLeaveCandidates', 'TooLowDelegationCountToDelegate', 'TooLowCandidateDelegationCountToDelegate', 'TooLowCandidateDelegationCountToLeaveCandidates', 'TooLowDelegationCountToLeaveDelegators', 'PendingCandidateRequestsDNE', 'PendingCandidateRequestAlreadyExists', 'PendingCandidateRequestNotDueYet', 'PendingDelegationRequestDNE', 'PendingDelegationRequestAlreadyExists', 'PendingDelegationRequestNotDueYet', 'CannotDelegateLessThanOrEqualToLowestBottomWhenFull', 'PendingDelegationRevoke']
    },
    /**
     * Lookup252: pallet_valve::pallet::Call<T>
     **/
    PalletValveCall: {
        _enum: {
            close_valve: 'Null',
            close_pallet_gate: {
                palletName: 'Bytes',
            },
            open_valve: 'Null',
            open_pallet_gate: {
                palletName: 'Bytes',
            },
            open_pallet_gates: 'Null',
            stop_scheduled_tasks: 'Null',
            start_scheduled_tasks: 'Null',
            stop_price_automation_tasks: 'Null',
            start_price_automation_tasks: 'Null'
        }
    },
    /**
     * Lookup253: pallet_valve::pallet::Error<T>
     **/
    PalletValveError: {
        _enum: ['ValveAlreadyClosed', 'ValveAlreadyOpen', 'InvalidCharacter', 'CannotCloseGate', 'ScheduledTasksAlreadyStopped', 'ScheduledTasksAlreadyRunnung', 'NotAllowed']
    },
    /**
     * Lookup254: pallet_identity::types::Registration<Balance, MaxJudgements, MaxAdditionalFields>
     **/
    PalletIdentityRegistration: {
        judgements: 'Vec<(u32,PalletIdentityJudgement)>',
        deposit: 'u128',
        info: 'PalletIdentityIdentityInfo'
    },
    /**
     * Lookup257: pallet_identity::types::Judgement<Balance>
     **/
    PalletIdentityJudgement: {
        _enum: {
            Unknown: 'Null',
            FeePaid: 'u128',
            Reasonable: 'Null',
            KnownGood: 'Null',
            OutOfDate: 'Null',
            LowQuality: 'Null',
            Erroneous: 'Null'
        }
    },
    /**
     * Lookup259: pallet_identity::types::IdentityInfo<FieldLimit>
     **/
    PalletIdentityIdentityInfo: {
        additional: 'Vec<(Data,Data)>',
        display: 'Data',
        legal: 'Data',
        web: 'Data',
        riot: 'Data',
        email: 'Data',
        pgpFingerprint: 'Option<[u8;20]>',
        image: 'Data',
        twitter: 'Data'
    },
    /**
     * Lookup298: pallet_identity::types::RegistrarInfo<Balance, sp_core::crypto::AccountId32>
     **/
    PalletIdentityRegistrarInfo: {
        account: 'AccountId32',
        fee: 'u128',
        fields: 'PalletIdentityBitFlags'
    },
    /**
     * Lookup299: pallet_identity::types::BitFlags<pallet_identity::types::IdentityField>
     **/
    PalletIdentityBitFlags: {
        _bitLength: 64,
        Display: 1,
        Legal: 2,
        Web: 4,
        Riot: 8,
        Email: 16,
        PgpFingerprint: 32,
        Image: 64,
        Twitter: 128
    },
    /**
     * Lookup300: pallet_identity::types::IdentityField
     **/
    PalletIdentityIdentityField: {
        _enum: ['__Unused0', 'Display', 'Legal', '__Unused3', 'Web', '__Unused5', '__Unused6', '__Unused7', 'Riot', '__Unused9', '__Unused10', '__Unused11', '__Unused12', '__Unused13', '__Unused14', '__Unused15', 'Email', '__Unused17', '__Unused18', '__Unused19', '__Unused20', '__Unused21', '__Unused22', '__Unused23', '__Unused24', '__Unused25', '__Unused26', '__Unused27', '__Unused28', '__Unused29', '__Unused30', '__Unused31', 'PgpFingerprint', '__Unused33', '__Unused34', '__Unused35', '__Unused36', '__Unused37', '__Unused38', '__Unused39', '__Unused40', '__Unused41', '__Unused42', '__Unused43', '__Unused44', '__Unused45', '__Unused46', '__Unused47', '__Unused48', '__Unused49', '__Unused50', '__Unused51', '__Unused52', '__Unused53', '__Unused54', '__Unused55', '__Unused56', '__Unused57', '__Unused58', '__Unused59', '__Unused60', '__Unused61', '__Unused62', '__Unused63', 'Image', '__Unused65', '__Unused66', '__Unused67', '__Unused68', '__Unused69', '__Unused70', '__Unused71', '__Unused72', '__Unused73', '__Unused74', '__Unused75', '__Unused76', '__Unused77', '__Unused78', '__Unused79', '__Unused80', '__Unused81', '__Unused82', '__Unused83', '__Unused84', '__Unused85', '__Unused86', '__Unused87', '__Unused88', '__Unused89', '__Unused90', '__Unused91', '__Unused92', '__Unused93', '__Unused94', '__Unused95', '__Unused96', '__Unused97', '__Unused98', '__Unused99', '__Unused100', '__Unused101', '__Unused102', '__Unused103', '__Unused104', '__Unused105', '__Unused106', '__Unused107', '__Unused108', '__Unused109', '__Unused110', '__Unused111', '__Unused112', '__Unused113', '__Unused114', '__Unused115', '__Unused116', '__Unused117', '__Unused118', '__Unused119', '__Unused120', '__Unused121', '__Unused122', '__Unused123', '__Unused124', '__Unused125', '__Unused126', '__Unused127', 'Twitter']
    },
    /**
     * Lookup302: pallet_identity::pallet::Call<T>
     **/
    PalletIdentityCall: {
        _enum: {
            add_registrar: {
                account: 'AccountId32',
            },
            set_identity: {
                info: 'PalletIdentityIdentityInfo',
            },
            set_subs: {
                subs: 'Vec<(AccountId32,Data)>',
            },
            clear_identity: 'Null',
            request_judgement: {
                regIndex: 'Compact<u32>',
                maxFee: 'Compact<u128>',
            },
            cancel_request: {
                regIndex: 'u32',
            },
            set_fee: {
                index: 'Compact<u32>',
                fee: 'Compact<u128>',
            },
            set_account_id: {
                _alias: {
                    new_: 'new',
                },
                index: 'Compact<u32>',
                new_: 'AccountId32',
            },
            set_fields: {
                index: 'Compact<u32>',
                fields: 'PalletIdentityBitFlags',
            },
            provide_judgement: {
                regIndex: 'Compact<u32>',
                target: 'MultiAddress',
                judgement: 'PalletIdentityJudgement',
            },
            kill_identity: {
                target: 'MultiAddress',
            },
            add_sub: {
                sub: 'MultiAddress',
                data: 'Data',
            },
            rename_sub: {
                sub: 'MultiAddress',
                data: 'Data',
            },
            remove_sub: {
                sub: 'MultiAddress',
            },
            quit_sub: 'Null'
        }
    },
    /**
     * Lookup304: pallet_identity::pallet::Error<T>
     **/
    PalletIdentityError: {
        _enum: ['TooManySubAccounts', 'NotFound', 'NotNamed', 'EmptyIndex', 'FeeChanged', 'NoIdentity', 'StickyJudgement', 'JudgementGiven', 'InvalidJudgement', 'InvalidIndex', 'InvalidTarget', 'TooManyFields', 'TooManyRegistrars', 'AlreadyClaimed', 'NotSub', 'NotOwned']
    },
    /**
     * Lookup307: pallet_proxy::ProxyDefinition<sp_core::crypto::AccountId32, turing_runtime::ProxyType, BlockNumber>
     **/
    PalletProxyProxyDefinition: {
        delegate: 'AccountId32',
        proxyType: 'TuringRuntimeProxyType',
        delay: 'u32'
    },
    /**
     * Lookup311: pallet_proxy::Announcement<sp_core::crypto::AccountId32, primitive_types::H256, BlockNumber>
     **/
    PalletProxyAnnouncement: {
        real: 'AccountId32',
        callHash: 'H256',
        height: 'u32'
    },
    /**
     * Lookup313: pallet_proxy::pallet::Call<T>
     **/
    PalletProxyCall: {
        _enum: {
            proxy: {
                real: 'AccountId32',
                forceProxyType: 'Option<TuringRuntimeProxyType>',
                call: 'Call',
            },
            add_proxy: {
                delegate: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                delay: 'u32',
            },
            remove_proxy: {
                delegate: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                delay: 'u32',
            },
            remove_proxies: 'Null',
            anonymous: {
                proxyType: 'TuringRuntimeProxyType',
                delay: 'u32',
                index: 'u16',
            },
            kill_anonymous: {
                spawner: 'AccountId32',
                proxyType: 'TuringRuntimeProxyType',
                index: 'u16',
                height: 'Compact<u32>',
                extIndex: 'Compact<u32>',
            },
            announce: {
                real: 'AccountId32',
                callHash: 'H256',
            },
            remove_announcement: {
                real: 'AccountId32',
                callHash: 'H256',
            },
            reject_announcement: {
                delegate: 'AccountId32',
                callHash: 'H256',
            },
            proxy_announced: {
                delegate: 'AccountId32',
                real: 'AccountId32',
                forceProxyType: 'Option<TuringRuntimeProxyType>',
                call: 'Call'
            }
        }
    },
    /**
     * Lookup316: pallet_utility::pallet::Call<T>
     **/
    PalletUtilityCall: {
        _enum: {
            batch: {
                calls: 'Vec<Call>',
            },
            as_derivative: {
                index: 'u16',
                call: 'Call',
            },
            batch_all: {
                calls: 'Vec<Call>',
            },
            dispatch_as: {
                asOrigin: 'TuringRuntimeOriginCaller',
                call: 'Call',
            },
            force_batch: {
                calls: 'Vec<Call>'
            }
        }
    },
    /**
     * Lookup318: turing_runtime::OriginCaller
     **/
    TuringRuntimeOriginCaller: {
        _enum: {
            system: 'FrameSupportDispatchRawOrigin',
            __Unused1: 'Null',
            __Unused2: 'Null',
            __Unused3: 'Null',
            __Unused4: 'Null',
            Void: 'SpCoreVoid',
            __Unused6: 'Null',
            __Unused7: 'Null',
            __Unused8: 'Null',
            __Unused9: 'Null',
            __Unused10: 'Null',
            __Unused11: 'Null',
            __Unused12: 'Null',
            __Unused13: 'Null',
            __Unused14: 'Null',
            __Unused15: 'Null',
            __Unused16: 'Null',
            __Unused17: 'Null',
            __Unused18: 'Null',
            __Unused19: 'Null',
            __Unused20: 'Null',
            __Unused21: 'Null',
            __Unused22: 'Null',
            __Unused23: 'Null',
            __Unused24: 'Null',
            __Unused25: 'Null',
            __Unused26: 'Null',
            __Unused27: 'Null',
            __Unused28: 'Null',
            __Unused29: 'Null',
            __Unused30: 'Null',
            __Unused31: 'Null',
            __Unused32: 'Null',
            __Unused33: 'Null',
            __Unused34: 'Null',
            __Unused35: 'Null',
            __Unused36: 'Null',
            __Unused37: 'Null',
            __Unused38: 'Null',
            __Unused39: 'Null',
            __Unused40: 'Null',
            PolkadotXcm: 'PalletXcmOrigin',
            CumulusXcm: 'CumulusPalletXcmOrigin',
            __Unused43: 'Null',
            __Unused44: 'Null',
            __Unused45: 'Null',
            __Unused46: 'Null',
            __Unused47: 'Null',
            __Unused48: 'Null',
            __Unused49: 'Null',
            __Unused50: 'Null',
            __Unused51: 'Null',
            Council: 'PalletCollectiveRawOrigin',
            TechnicalCommittee: 'PalletCollectiveRawOrigin'
        }
    },
    /**
     * Lookup319: frame_support::dispatch::RawOrigin<sp_core::crypto::AccountId32>
     **/
    FrameSupportDispatchRawOrigin: {
        _enum: {
            Root: 'Null',
            Signed: 'AccountId32',
            None: 'Null'
        }
    },
    /**
     * Lookup320: pallet_xcm::pallet::Origin
     **/
    PalletXcmOrigin: {
        _enum: {
            Xcm: 'XcmV1MultiLocation',
            Response: 'XcmV1MultiLocation'
        }
    },
    /**
     * Lookup321: cumulus_pallet_xcm::pallet::Origin
     **/
    CumulusPalletXcmOrigin: {
        _enum: {
            Relay: 'Null',
            SiblingParachain: 'u32'
        }
    },
    /**
     * Lookup322: pallet_collective::RawOrigin<sp_core::crypto::AccountId32, I>
     **/
    PalletCollectiveRawOrigin: {
        _enum: {
            Members: '(u32,u32)',
            Member: 'AccountId32',
            _Phantom: 'Null'
        }
    },
    /**
     * Lookup324: sp_core::Void
     **/
    SpCoreVoid: 'Null',
    /**
     * Lookup325: cumulus_pallet_xcmp_queue::pallet::Call<T>
     **/
    CumulusPalletXcmpQueueCall: {
        _enum: {
            service_overweight: {
                index: 'u64',
                weightLimit: 'u64',
            },
            suspend_xcm_execution: 'Null',
            resume_xcm_execution: 'Null',
            update_suspend_threshold: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u32',
            },
            update_drop_threshold: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u32',
            },
            update_resume_threshold: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u32',
            },
            update_threshold_weight: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u64',
            },
            update_weight_restrict_decay: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u64',
            },
            update_xcmp_max_individual_weight: {
                _alias: {
                    new_: 'new',
                },
                new_: 'u64'
            }
        }
    },
    /**
     * Lookup326: pallet_xcm::pallet::Call<T>
     **/
    PalletXcmCall: {
        _enum: {
            send: {
                dest: 'XcmVersionedMultiLocation',
                message: 'XcmVersionedXcm',
            },
            teleport_assets: {
                dest: 'XcmVersionedMultiLocation',
                beneficiary: 'XcmVersionedMultiLocation',
                assets: 'XcmVersionedMultiAssets',
                feeAssetItem: 'u32',
            },
            reserve_transfer_assets: {
                dest: 'XcmVersionedMultiLocation',
                beneficiary: 'XcmVersionedMultiLocation',
                assets: 'XcmVersionedMultiAssets',
                feeAssetItem: 'u32',
            },
            execute: {
                message: 'XcmVersionedXcm',
                maxWeight: 'u64',
            },
            force_xcm_version: {
                location: 'XcmV1MultiLocation',
                xcmVersion: 'u32',
            },
            force_default_xcm_version: {
                maybeXcmVersion: 'Option<u32>',
            },
            force_subscribe_version_notify: {
                location: 'XcmVersionedMultiLocation',
            },
            force_unsubscribe_version_notify: {
                location: 'XcmVersionedMultiLocation',
            },
            limited_reserve_transfer_assets: {
                dest: 'XcmVersionedMultiLocation',
                beneficiary: 'XcmVersionedMultiLocation',
                assets: 'XcmVersionedMultiAssets',
                feeAssetItem: 'u32',
                weightLimit: 'XcmV2WeightLimit',
            },
            limited_teleport_assets: {
                dest: 'XcmVersionedMultiLocation',
                beneficiary: 'XcmVersionedMultiLocation',
                assets: 'XcmVersionedMultiAssets',
                feeAssetItem: 'u32',
                weightLimit: 'XcmV2WeightLimit'
            }
        }
    },
    /**
     * Lookup327: xcm::VersionedXcm<Call>
     **/
    XcmVersionedXcm: {
        _enum: {
            V0: 'XcmV0Xcm',
            V1: 'XcmV1Xcm',
            V2: 'XcmV2Xcm'
        }
    },
    /**
     * Lookup328: xcm::v0::Xcm<Call>
     **/
    XcmV0Xcm: {
        _enum: {
            WithdrawAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                effects: 'Vec<XcmV0Order>',
            },
            ReserveAssetDeposit: {
                assets: 'Vec<XcmV0MultiAsset>',
                effects: 'Vec<XcmV0Order>',
            },
            TeleportAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                effects: 'Vec<XcmV0Order>',
            },
            QueryResponse: {
                queryId: 'Compact<u64>',
                response: 'XcmV0Response',
            },
            TransferAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                dest: 'XcmV0MultiLocation',
            },
            TransferReserveAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                dest: 'XcmV0MultiLocation',
                effects: 'Vec<XcmV0Order>',
            },
            Transact: {
                originType: 'XcmV0OriginKind',
                requireWeightAtMost: 'u64',
                call: 'XcmDoubleEncoded',
            },
            HrmpNewChannelOpenRequest: {
                sender: 'Compact<u32>',
                maxMessageSize: 'Compact<u32>',
                maxCapacity: 'Compact<u32>',
            },
            HrmpChannelAccepted: {
                recipient: 'Compact<u32>',
            },
            HrmpChannelClosing: {
                initiator: 'Compact<u32>',
                sender: 'Compact<u32>',
                recipient: 'Compact<u32>',
            },
            RelayedFrom: {
                who: 'XcmV0MultiLocation',
                message: 'XcmV0Xcm'
            }
        }
    },
    /**
     * Lookup330: xcm::v0::order::Order<Call>
     **/
    XcmV0Order: {
        _enum: {
            Null: 'Null',
            DepositAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                dest: 'XcmV0MultiLocation',
            },
            DepositReserveAsset: {
                assets: 'Vec<XcmV0MultiAsset>',
                dest: 'XcmV0MultiLocation',
                effects: 'Vec<XcmV0Order>',
            },
            ExchangeAsset: {
                give: 'Vec<XcmV0MultiAsset>',
                receive: 'Vec<XcmV0MultiAsset>',
            },
            InitiateReserveWithdraw: {
                assets: 'Vec<XcmV0MultiAsset>',
                reserve: 'XcmV0MultiLocation',
                effects: 'Vec<XcmV0Order>',
            },
            InitiateTeleport: {
                assets: 'Vec<XcmV0MultiAsset>',
                dest: 'XcmV0MultiLocation',
                effects: 'Vec<XcmV0Order>',
            },
            QueryHolding: {
                queryId: 'Compact<u64>',
                dest: 'XcmV0MultiLocation',
                assets: 'Vec<XcmV0MultiAsset>',
            },
            BuyExecution: {
                fees: 'XcmV0MultiAsset',
                weight: 'u64',
                debt: 'u64',
                haltOnError: 'bool',
                xcm: 'Vec<XcmV0Xcm>'
            }
        }
    },
    /**
     * Lookup332: xcm::v0::Response
     **/
    XcmV0Response: {
        _enum: {
            Assets: 'Vec<XcmV0MultiAsset>'
        }
    },
    /**
     * Lookup333: xcm::v1::Xcm<Call>
     **/
    XcmV1Xcm: {
        _enum: {
            WithdrawAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                effects: 'Vec<XcmV1Order>',
            },
            ReserveAssetDeposited: {
                assets: 'XcmV1MultiassetMultiAssets',
                effects: 'Vec<XcmV1Order>',
            },
            ReceiveTeleportedAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                effects: 'Vec<XcmV1Order>',
            },
            QueryResponse: {
                queryId: 'Compact<u64>',
                response: 'XcmV1Response',
            },
            TransferAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                beneficiary: 'XcmV1MultiLocation',
            },
            TransferReserveAsset: {
                assets: 'XcmV1MultiassetMultiAssets',
                dest: 'XcmV1MultiLocation',
                effects: 'Vec<XcmV1Order>',
            },
            Transact: {
                originType: 'XcmV0OriginKind',
                requireWeightAtMost: 'u64',
                call: 'XcmDoubleEncoded',
            },
            HrmpNewChannelOpenRequest: {
                sender: 'Compact<u32>',
                maxMessageSize: 'Compact<u32>',
                maxCapacity: 'Compact<u32>',
            },
            HrmpChannelAccepted: {
                recipient: 'Compact<u32>',
            },
            HrmpChannelClosing: {
                initiator: 'Compact<u32>',
                sender: 'Compact<u32>',
                recipient: 'Compact<u32>',
            },
            RelayedFrom: {
                who: 'XcmV1MultilocationJunctions',
                message: 'XcmV1Xcm',
            },
            SubscribeVersion: {
                queryId: 'Compact<u64>',
                maxResponseWeight: 'Compact<u64>',
            },
            UnsubscribeVersion: 'Null'
        }
    },
    /**
     * Lookup335: xcm::v1::order::Order<Call>
     **/
    XcmV1Order: {
        _enum: {
            Noop: 'Null',
            DepositAsset: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                maxAssets: 'u32',
                beneficiary: 'XcmV1MultiLocation',
            },
            DepositReserveAsset: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                maxAssets: 'u32',
                dest: 'XcmV1MultiLocation',
                effects: 'Vec<XcmV1Order>',
            },
            ExchangeAsset: {
                give: 'XcmV1MultiassetMultiAssetFilter',
                receive: 'XcmV1MultiassetMultiAssets',
            },
            InitiateReserveWithdraw: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                reserve: 'XcmV1MultiLocation',
                effects: 'Vec<XcmV1Order>',
            },
            InitiateTeleport: {
                assets: 'XcmV1MultiassetMultiAssetFilter',
                dest: 'XcmV1MultiLocation',
                effects: 'Vec<XcmV1Order>',
            },
            QueryHolding: {
                queryId: 'Compact<u64>',
                dest: 'XcmV1MultiLocation',
                assets: 'XcmV1MultiassetMultiAssetFilter',
            },
            BuyExecution: {
                fees: 'XcmV1MultiAsset',
                weight: 'u64',
                debt: 'u64',
                haltOnError: 'bool',
                instructions: 'Vec<XcmV1Xcm>'
            }
        }
    },
    /**
     * Lookup337: xcm::v1::Response
     **/
    XcmV1Response: {
        _enum: {
            Assets: 'XcmV1MultiassetMultiAssets',
            Version: 'u32'
        }
    },
    /**
     * Lookup351: cumulus_pallet_dmp_queue::pallet::Call<T>
     **/
    CumulusPalletDmpQueueCall: {
        _enum: {
            service_overweight: {
                index: 'u64',
                weightLimit: 'u64'
            }
        }
    },
    /**
     * Lookup352: orml_xtokens::module::Call<T>
     **/
    OrmlXtokensModuleCall: {
        _enum: {
            transfer: {
                currencyId: 'u32',
                amount: 'u128',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64',
            },
            transfer_multiasset: {
                asset: 'XcmVersionedMultiAsset',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64',
            },
            transfer_with_fee: {
                currencyId: 'u32',
                amount: 'u128',
                fee: 'u128',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64',
            },
            transfer_multiasset_with_fee: {
                asset: 'XcmVersionedMultiAsset',
                fee: 'XcmVersionedMultiAsset',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64',
            },
            transfer_multicurrencies: {
                currencies: 'Vec<(u32,u128)>',
                feeItem: 'u32',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64',
            },
            transfer_multiassets: {
                assets: 'XcmVersionedMultiAssets',
                feeItem: 'u32',
                dest: 'XcmVersionedMultiLocation',
                destWeight: 'u64'
            }
        }
    },
    /**
     * Lookup353: xcm::VersionedMultiAsset
     **/
    XcmVersionedMultiAsset: {
        _enum: {
            V0: 'XcmV0MultiAsset',
            V1: 'XcmV1MultiAsset'
        }
    },
    /**
     * Lookup356: pallet_sudo::pallet::Call<T>
     **/
    PalletSudoCall: {
        _enum: {
            sudo: {
                call: 'Call',
            },
            sudo_unchecked_weight: {
                call: 'Call',
                weight: 'u64',
            },
            set_key: {
                _alias: {
                    new_: 'new',
                },
                new_: 'MultiAddress',
            },
            sudo_as: {
                who: 'MultiAddress',
                call: 'Call'
            }
        }
    },
    /**
     * Lookup357: pallet_treasury::pallet::Call<T, I>
     **/
    PalletTreasuryCall: {
        _enum: {
            propose_spend: {
                value: 'Compact<u128>',
                beneficiary: 'MultiAddress',
            },
            reject_proposal: {
                proposalId: 'Compact<u32>',
            },
            approve_proposal: {
                proposalId: 'Compact<u32>',
            },
            spend: {
                amount: 'Compact<u128>',
                beneficiary: 'MultiAddress',
            },
            remove_approval: {
                proposalId: 'Compact<u32>'
            }
        }
    },
    /**
     * Lookup358: pallet_collective::pallet::Call<T, I>
     **/
    PalletCollectiveCall: {
        _enum: {
            set_members: {
                newMembers: 'Vec<AccountId32>',
                prime: 'Option<AccountId32>',
                oldCount: 'u32',
            },
            execute: {
                proposal: 'Call',
                lengthBound: 'Compact<u32>',
            },
            propose: {
                threshold: 'Compact<u32>',
                proposal: 'Call',
                lengthBound: 'Compact<u32>',
            },
            vote: {
                proposal: 'H256',
                index: 'Compact<u32>',
                approve: 'bool',
            },
            close: {
                proposalHash: 'H256',
                index: 'Compact<u32>',
                proposalWeightBound: 'Compact<u64>',
                lengthBound: 'Compact<u32>',
            },
            disapprove_proposal: {
                proposalHash: 'H256'
            }
        }
    },
    /**
     * Lookup360: pallet_membership::pallet::Call<T, I>
     **/
    PalletMembershipCall: {
        _enum: {
            add_member: {
                who: 'AccountId32',
            },
            remove_member: {
                who: 'AccountId32',
            },
            swap_member: {
                remove: 'AccountId32',
                add: 'AccountId32',
            },
            reset_members: {
                members: 'Vec<AccountId32>',
            },
            change_key: {
                _alias: {
                    new_: 'new',
                },
                new_: 'AccountId32',
            },
            set_prime: {
                who: 'AccountId32',
            },
            clear_prime: 'Null'
        }
    },
    /**
     * Lookup361: pallet_bounties::pallet::Call<T, I>
     **/
    PalletBountiesCall: {
        _enum: {
            propose_bounty: {
                value: 'Compact<u128>',
                description: 'Bytes',
            },
            approve_bounty: {
                bountyId: 'Compact<u32>',
            },
            propose_curator: {
                bountyId: 'Compact<u32>',
                curator: 'MultiAddress',
                fee: 'Compact<u128>',
            },
            unassign_curator: {
                bountyId: 'Compact<u32>',
            },
            accept_curator: {
                bountyId: 'Compact<u32>',
            },
            award_bounty: {
                bountyId: 'Compact<u32>',
                beneficiary: 'MultiAddress',
            },
            claim_bounty: {
                bountyId: 'Compact<u32>',
            },
            close_bounty: {
                bountyId: 'Compact<u32>',
            },
            extend_bounty_expiry: {
                bountyId: 'Compact<u32>',
                remark: 'Bytes'
            }
        }
    },
    /**
     * Lookup362: pallet_preimage::pallet::Call<T>
     **/
    PalletPreimageCall: {
        _enum: {
            note_preimage: {
                bytes: 'Bytes',
            },
            unnote_preimage: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256',
            },
            request_preimage: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256',
            },
            unrequest_preimage: {
                _alias: {
                    hash_: 'hash',
                },
                hash_: 'H256'
            }
        }
    },
    /**
     * Lookup363: pallet_scheduler::pallet::Call<T>
     **/
    PalletSchedulerCall: {
        _enum: {
            schedule: {
                when: 'u32',
                maybePeriodic: 'Option<(u32,u32)>',
                priority: 'u8',
                call: 'FrameSupportScheduleMaybeHashed',
            },
            cancel: {
                when: 'u32',
                index: 'u32',
            },
            schedule_named: {
                id: 'Bytes',
                when: 'u32',
                maybePeriodic: 'Option<(u32,u32)>',
                priority: 'u8',
                call: 'FrameSupportScheduleMaybeHashed',
            },
            cancel_named: {
                id: 'Bytes',
            },
            schedule_after: {
                after: 'u32',
                maybePeriodic: 'Option<(u32,u32)>',
                priority: 'u8',
                call: 'FrameSupportScheduleMaybeHashed',
            },
            schedule_named_after: {
                id: 'Bytes',
                after: 'u32',
                maybePeriodic: 'Option<(u32,u32)>',
                priority: 'u8',
                call: 'FrameSupportScheduleMaybeHashed'
            }
        }
    },
    /**
     * Lookup365: frame_support::traits::schedule::MaybeHashed<turing_runtime::Call, primitive_types::H256>
     **/
    FrameSupportScheduleMaybeHashed: {
        _enum: {
            Value: 'Call',
            Hash: 'H256'
        }
    },
    /**
     * Lookup366: pallet_democracy::pallet::Call<T>
     **/
    PalletDemocracyCall: {
        _enum: {
            propose: {
                proposalHash: 'H256',
                value: 'Compact<u128>',
            },
            second: {
                proposal: 'Compact<u32>',
                secondsUpperBound: 'Compact<u32>',
            },
            vote: {
                refIndex: 'Compact<u32>',
                vote: 'PalletDemocracyVoteAccountVote',
            },
            emergency_cancel: {
                refIndex: 'u32',
            },
            external_propose: {
                proposalHash: 'H256',
            },
            external_propose_majority: {
                proposalHash: 'H256',
            },
            external_propose_default: {
                proposalHash: 'H256',
            },
            fast_track: {
                proposalHash: 'H256',
                votingPeriod: 'u32',
                delay: 'u32',
            },
            veto_external: {
                proposalHash: 'H256',
            },
            cancel_referendum: {
                refIndex: 'Compact<u32>',
            },
            cancel_queued: {
                which: 'u32',
            },
            delegate: {
                to: 'AccountId32',
                conviction: 'PalletDemocracyConviction',
                balance: 'u128',
            },
            undelegate: 'Null',
            clear_public_proposals: 'Null',
            note_preimage: {
                encodedProposal: 'Bytes',
            },
            note_preimage_operational: {
                encodedProposal: 'Bytes',
            },
            note_imminent_preimage: {
                encodedProposal: 'Bytes',
            },
            note_imminent_preimage_operational: {
                encodedProposal: 'Bytes',
            },
            reap_preimage: {
                proposalHash: 'H256',
                proposalLenUpperBound: 'Compact<u32>',
            },
            unlock: {
                target: 'AccountId32',
            },
            remove_vote: {
                index: 'u32',
            },
            remove_other_vote: {
                target: 'AccountId32',
                index: 'u32',
            },
            enact_proposal: {
                proposalHash: 'H256',
                index: 'u32',
            },
            blacklist: {
                proposalHash: 'H256',
                maybeRefIndex: 'Option<u32>',
            },
            cancel_proposal: {
                propIndex: 'Compact<u32>'
            }
        }
    },
    /**
     * Lookup367: pallet_democracy::conviction::Conviction
     **/
    PalletDemocracyConviction: {
        _enum: ['None', 'Locked1x', 'Locked2x', 'Locked3x', 'Locked4x', 'Locked5x', 'Locked6x']
    },
    /**
     * Lookup368: pallet_multisig::pallet::Call<T>
     **/
    PalletMultisigCall: {
        _enum: {
            as_multi_threshold_1: {
                otherSignatories: 'Vec<AccountId32>',
                call: 'Call',
            },
            as_multi: {
                threshold: 'u16',
                otherSignatories: 'Vec<AccountId32>',
                maybeTimepoint: 'Option<PalletMultisigTimepoint>',
                call: 'WrapperKeepOpaque<Call>',
                storeCall: 'bool',
                maxWeight: 'u64',
            },
            approve_as_multi: {
                threshold: 'u16',
                otherSignatories: 'Vec<AccountId32>',
                maybeTimepoint: 'Option<PalletMultisigTimepoint>',
                callHash: '[u8;32]',
                maxWeight: 'u64',
            },
            cancel_as_multi: {
                threshold: 'u16',
                otherSignatories: 'Vec<AccountId32>',
                timepoint: 'PalletMultisigTimepoint',
                callHash: '[u8;32]'
            }
        }
    },
    /**
     * Lookup371: pallet_automation_time::pallet::Call<T>
     **/
    PalletAutomationTimeCall: {
        _enum: {
            schedule_notify_task: {
                providedId: 'Bytes',
                executionTimes: 'Vec<u64>',
                message: 'Bytes',
            },
            schedule_native_transfer_task: {
                providedId: 'Bytes',
                executionTimes: 'Vec<u64>',
                recipientId: 'AccountId32',
                amount: 'Compact<u128>',
            },
            schedule_xcmp_task: {
                providedId: 'Bytes',
                schedule: 'PalletAutomationTimeScheduleParam',
                paraId: 'u32',
                currencyId: 'u32',
                encodedCall: 'Bytes',
                encodedCallWeight: 'u64',
            },
            schedule_auto_compound_delegated_stake_task: {
                executionTime: 'u64',
                frequency: 'u64',
                collatorId: 'AccountId32',
                accountMinimum: 'u128',
            },
            schedule_dynamic_dispatch_task: {
                providedId: 'Bytes',
                schedule: 'PalletAutomationTimeScheduleParam',
                call: 'Call',
            },
            cancel_task: {
                taskId: 'H256',
            },
            force_cancel_task: {
                ownerId: 'AccountId32',
                taskId: 'H256'
            }
        }
    },
    /**
     * Lookup373: pallet_automation_time::types::ScheduleParam
     **/
    PalletAutomationTimeScheduleParam: {
        _enum: {
            Fixed: {
                executionTimes: 'Vec<u64>',
            },
            Recurring: {
                nextExecutionTime: 'u64',
                frequency: 'u64'
            }
        }
    },
    /**
     * Lookup374: pallet_xcmp_handler::pallet::Call<T>
     **/
    PalletXcmpHandlerCall: {
        _enum: {
            add_chain_currency_data: {
                paraId: 'u32',
                currencyId: 'u32',
                xcmData: 'PalletXcmpHandlerXcmCurrencyData',
            },
            remove_chain_currency_data: {
                paraId: 'u32',
                currencyId: 'u32'
            }
        }
    },
    /**
     * Lookup375: pallet_xcmp_handler::pallet::XcmCurrencyData
     **/
    PalletXcmpHandlerXcmCurrencyData: {
        native: 'bool',
        feePerSecond: 'u128',
        instructionWeight: 'u64'
    },
    /**
     * Lookup376: pallet_automation_price::pallet::Call<T>
     **/
    PalletAutomationPriceCall: {
        _enum: {
            schedule_transfer_task: {
                providedId: 'Bytes',
                asset: 'Bytes',
                direction: 'PalletAutomationPriceDirection',
                triggerPercentage: 'u128',
                recipient: 'AccountId32',
                amount: 'Compact<u128>',
            },
            add_asset: {
                asset: 'Bytes',
                targetPrice: 'u128',
                upperBound: 'u16',
                lowerBound: 'u8',
                assetOwner: 'AccountId32',
                expirationPeriod: 'u64',
            },
            asset_price_update: {
                asset: 'Bytes',
                value: 'u128',
            },
            delete_asset: {
                asset: 'Bytes'
            }
        }
    },
    /**
     * Lookup377: pallet_automation_price::pallet::Direction
     **/
    PalletAutomationPriceDirection: {
        _enum: ['Up', 'Down']
    },
    /**
     * Lookup378: pallet_proxy::pallet::Error<T>
     **/
    PalletProxyError: {
        _enum: ['TooMany', 'NotFound', 'NotProxy', 'Unproxyable', 'Duplicate', 'NoPermission', 'Unannounced', 'NoSelfProxy']
    },
    /**
     * Lookup379: pallet_utility::pallet::Error<T>
     **/
    PalletUtilityError: {
        _enum: ['TooManyCalls']
    },
    /**
     * Lookup381: cumulus_pallet_xcmp_queue::InboundChannelDetails
     **/
    CumulusPalletXcmpQueueInboundChannelDetails: {
        sender: 'u32',
        state: 'CumulusPalletXcmpQueueInboundState',
        messageMetadata: 'Vec<(u32,PolkadotParachainPrimitivesXcmpMessageFormat)>'
    },
    /**
     * Lookup382: cumulus_pallet_xcmp_queue::InboundState
     **/
    CumulusPalletXcmpQueueInboundState: {
        _enum: ['Ok', 'Suspended']
    },
    /**
     * Lookup385: polkadot_parachain::primitives::XcmpMessageFormat
     **/
    PolkadotParachainPrimitivesXcmpMessageFormat: {
        _enum: ['ConcatenatedVersionedXcm', 'ConcatenatedEncodedBlob', 'Signals']
    },
    /**
     * Lookup388: cumulus_pallet_xcmp_queue::OutboundChannelDetails
     **/
    CumulusPalletXcmpQueueOutboundChannelDetails: {
        recipient: 'u32',
        state: 'CumulusPalletXcmpQueueOutboundState',
        signalsExist: 'bool',
        firstIndex: 'u16',
        lastIndex: 'u16'
    },
    /**
     * Lookup389: cumulus_pallet_xcmp_queue::OutboundState
     **/
    CumulusPalletXcmpQueueOutboundState: {
        _enum: ['Ok', 'Suspended']
    },
    /**
     * Lookup391: cumulus_pallet_xcmp_queue::QueueConfigData
     **/
    CumulusPalletXcmpQueueQueueConfigData: {
        suspendThreshold: 'u32',
        dropThreshold: 'u32',
        resumeThreshold: 'u32',
        thresholdWeight: 'u64',
        weightRestrictDecay: 'u64',
        xcmpMaxIndividualWeight: 'u64'
    },
    /**
     * Lookup393: cumulus_pallet_xcmp_queue::pallet::Error<T>
     **/
    CumulusPalletXcmpQueueError: {
        _enum: ['FailedToSend', 'BadXcmOrigin', 'BadXcm', 'BadOverweightIndex', 'WeightOverLimit']
    },
    /**
     * Lookup394: pallet_xcm::pallet::Error<T>
     **/
    PalletXcmError: {
        _enum: ['Unreachable', 'SendFailure', 'Filtered', 'UnweighableMessage', 'DestinationNotInvertible', 'Empty', 'CannotReanchor', 'TooManyAssets', 'InvalidOrigin', 'BadVersion', 'BadLocation', 'NoSubscription', 'AlreadySubscribed']
    },
    /**
     * Lookup395: cumulus_pallet_xcm::pallet::Error<T>
     **/
    CumulusPalletXcmError: 'Null',
    /**
     * Lookup396: cumulus_pallet_dmp_queue::ConfigData
     **/
    CumulusPalletDmpQueueConfigData: {
        maxIndividual: 'u64'
    },
    /**
     * Lookup397: cumulus_pallet_dmp_queue::PageIndexData
     **/
    CumulusPalletDmpQueuePageIndexData: {
        beginUsed: 'u32',
        endUsed: 'u32',
        overweightCount: 'u64'
    },
    /**
     * Lookup400: cumulus_pallet_dmp_queue::pallet::Error<T>
     **/
    CumulusPalletDmpQueueError: {
        _enum: ['Unknown', 'OverLimit']
    },
    /**
     * Lookup401: orml_xtokens::module::Error<T>
     **/
    OrmlXtokensModuleError: {
        _enum: ['AssetHasNoReserve', 'NotCrossChainTransfer', 'InvalidDest', 'NotCrossChainTransferableCurrency', 'UnweighableMessage', 'XcmExecutionFailed', 'CannotReanchor', 'InvalidAncestry', 'InvalidAsset', 'DestinationNotInvertible', 'BadVersion', 'DistinctReserveForAssetAndFee', 'ZeroFee', 'ZeroAmount', 'TooManyAssetsBeingSent', 'AssetIndexNonExistent', 'FeeNotEnough', 'NotSupportedMultiLocation', 'MinXcmFeeNotDefined']
    },
    /**
     * Lookup404: orml_unknown_tokens::module::Error<T>
     **/
    OrmlUnknownTokensModuleError: {
        _enum: ['BalanceTooLow', 'BalanceOverflow', 'UnhandledAsset']
    },
    /**
     * Lookup405: pallet_sudo::pallet::Error<T>
     **/
    PalletSudoError: {
        _enum: ['RequireSudo']
    },
    /**
     * Lookup406: pallet_treasury::Proposal<sp_core::crypto::AccountId32, Balance>
     **/
    PalletTreasuryProposal: {
        proposer: 'AccountId32',
        value: 'u128',
        beneficiary: 'AccountId32',
        bond: 'u128'
    },
    /**
     * Lookup409: frame_support::PalletId
     **/
    FrameSupportPalletId: '[u8;8]',
    /**
     * Lookup410: pallet_treasury::pallet::Error<T, I>
     **/
    PalletTreasuryError: {
        _enum: ['InsufficientProposersBalance', 'InvalidIndex', 'TooManyApprovals', 'InsufficientPermission', 'ProposalNotApproved']
    },
    /**
     * Lookup412: pallet_collective::Votes<sp_core::crypto::AccountId32, BlockNumber>
     **/
    PalletCollectiveVotes: {
        index: 'u32',
        threshold: 'u32',
        ayes: 'Vec<AccountId32>',
        nays: 'Vec<AccountId32>',
        end: 'u32'
    },
    /**
     * Lookup413: pallet_collective::pallet::Error<T, I>
     **/
    PalletCollectiveError: {
        _enum: ['NotMember', 'DuplicateProposal', 'ProposalMissing', 'WrongIndex', 'DuplicateVote', 'AlreadyInitialized', 'TooEarly', 'TooManyProposals', 'WrongProposalWeight', 'WrongProposalLength']
    },
    /**
     * Lookup416: pallet_membership::pallet::Error<T, I>
     **/
    PalletMembershipError: {
        _enum: ['AlreadyMember', 'NotMember', 'TooManyMembers']
    },
    /**
     * Lookup417: pallet_bounties::Bounty<sp_core::crypto::AccountId32, Balance, BlockNumber>
     **/
    PalletBountiesBounty: {
        proposer: 'AccountId32',
        value: 'u128',
        fee: 'u128',
        curatorDeposit: 'u128',
        bond: 'u128',
        status: 'PalletBountiesBountyStatus'
    },
    /**
     * Lookup418: pallet_bounties::BountyStatus<sp_core::crypto::AccountId32, BlockNumber>
     **/
    PalletBountiesBountyStatus: {
        _enum: {
            Proposed: 'Null',
            Approved: 'Null',
            Funded: 'Null',
            CuratorProposed: {
                curator: 'AccountId32',
            },
            Active: {
                curator: 'AccountId32',
                updateDue: 'u32',
            },
            PendingPayout: {
                curator: 'AccountId32',
                beneficiary: 'AccountId32',
                unlockAt: 'u32'
            }
        }
    },
    /**
     * Lookup420: pallet_bounties::pallet::Error<T, I>
     **/
    PalletBountiesError: {
        _enum: ['InsufficientProposersBalance', 'InvalidIndex', 'ReasonTooBig', 'UnexpectedStatus', 'RequireCurator', 'InvalidValue', 'InvalidFee', 'PendingPayout', 'Premature', 'HasActiveChildBounty', 'TooManyQueued']
    },
    /**
     * Lookup421: pallet_preimage::RequestStatus<sp_core::crypto::AccountId32, Balance>
     **/
    PalletPreimageRequestStatus: {
        _enum: {
            Unrequested: 'Option<(AccountId32,u128)>',
            Requested: 'u32'
        }
    },
    /**
     * Lookup425: pallet_preimage::pallet::Error<T>
     **/
    PalletPreimageError: {
        _enum: ['TooLarge', 'AlreadyNoted', 'NotAuthorized', 'NotNoted', 'Requested', 'NotRequested']
    },
    /**
     * Lookup428: pallet_scheduler::ScheduledV3<frame_support::traits::schedule::MaybeHashed<turing_runtime::Call, primitive_types::H256>, BlockNumber, turing_runtime::OriginCaller, sp_core::crypto::AccountId32>
     **/
    PalletSchedulerScheduledV3: {
        maybeId: 'Option<Bytes>',
        priority: 'u8',
        call: 'FrameSupportScheduleMaybeHashed',
        maybePeriodic: 'Option<(u32,u32)>',
        origin: 'TuringRuntimeOriginCaller'
    },
    /**
     * Lookup429: pallet_scheduler::pallet::Error<T>
     **/
    PalletSchedulerError: {
        _enum: ['FailedToSchedule', 'NotFound', 'TargetBlockNumberInPast', 'RescheduleNoChange']
    },
    /**
     * Lookup433: pallet_democracy::PreimageStatus<sp_core::crypto::AccountId32, Balance, BlockNumber>
     **/
    PalletDemocracyPreimageStatus: {
        _enum: {
            Missing: 'u32',
            Available: {
                data: 'Bytes',
                provider: 'AccountId32',
                deposit: 'u128',
                since: 'u32',
                expiry: 'Option<u32>'
            }
        }
    },
    /**
     * Lookup434: pallet_democracy::types::ReferendumInfo<BlockNumber, primitive_types::H256, Balance>
     **/
    PalletDemocracyReferendumInfo: {
        _enum: {
            Ongoing: 'PalletDemocracyReferendumStatus',
            Finished: {
                approved: 'bool',
                end: 'u32'
            }
        }
    },
    /**
     * Lookup435: pallet_democracy::types::ReferendumStatus<BlockNumber, primitive_types::H256, Balance>
     **/
    PalletDemocracyReferendumStatus: {
        end: 'u32',
        proposalHash: 'H256',
        threshold: 'PalletDemocracyVoteThreshold',
        delay: 'u32',
        tally: 'PalletDemocracyTally'
    },
    /**
     * Lookup436: pallet_democracy::types::Tally<Balance>
     **/
    PalletDemocracyTally: {
        ayes: 'u128',
        nays: 'u128',
        turnout: 'u128'
    },
    /**
     * Lookup437: pallet_democracy::vote::Voting<Balance, sp_core::crypto::AccountId32, BlockNumber>
     **/
    PalletDemocracyVoteVoting: {
        _enum: {
            Direct: {
                votes: 'Vec<(u32,PalletDemocracyVoteAccountVote)>',
                delegations: 'PalletDemocracyDelegations',
                prior: 'PalletDemocracyVotePriorLock',
            },
            Delegating: {
                balance: 'u128',
                target: 'AccountId32',
                conviction: 'PalletDemocracyConviction',
                delegations: 'PalletDemocracyDelegations',
                prior: 'PalletDemocracyVotePriorLock'
            }
        }
    },
    /**
     * Lookup440: pallet_democracy::types::Delegations<Balance>
     **/
    PalletDemocracyDelegations: {
        votes: 'u128',
        capital: 'u128'
    },
    /**
     * Lookup441: pallet_democracy::vote::PriorLock<BlockNumber, Balance>
     **/
    PalletDemocracyVotePriorLock: '(u32,u128)',
    /**
     * Lookup444: pallet_democracy::Releases
     **/
    PalletDemocracyReleases: {
        _enum: ['V1']
    },
    /**
     * Lookup445: pallet_democracy::pallet::Error<T>
     **/
    PalletDemocracyError: {
        _enum: ['ValueLow', 'ProposalMissing', 'AlreadyCanceled', 'DuplicateProposal', 'ProposalBlacklisted', 'NotSimpleMajority', 'InvalidHash', 'NoProposal', 'AlreadyVetoed', 'DuplicatePreimage', 'NotImminent', 'TooEarly', 'Imminent', 'PreimageMissing', 'ReferendumInvalid', 'PreimageInvalid', 'NoneWaiting', 'NotVoter', 'NoPermission', 'AlreadyDelegating', 'InsufficientFunds', 'NotDelegating', 'VotesExist', 'InstantNotAllowed', 'Nonsense', 'WrongUpperBound', 'MaxVotesReached', 'TooManyProposals', 'VotingPeriodLow']
    },
    /**
     * Lookup447: pallet_multisig::Multisig<BlockNumber, Balance, sp_core::crypto::AccountId32>
     **/
    PalletMultisigMultisig: {
        when: 'PalletMultisigTimepoint',
        deposit: 'u128',
        depositor: 'AccountId32',
        approvals: 'Vec<AccountId32>'
    },
    /**
     * Lookup449: pallet_multisig::pallet::Error<T>
     **/
    PalletMultisigError: {
        _enum: ['MinimumThreshold', 'AlreadyApproved', 'NoApprovalsNeeded', 'TooFewSignatories', 'TooManySignatories', 'SignatoriesOutOfOrder', 'SenderInSignatories', 'NotFound', 'NotOwner', 'NoTimepoint', 'WrongTimepoint', 'UnexpectedTimepoint', 'MaxWeightTooLow', 'AlreadyStored']
    },
    /**
     * Lookup450: pallet_automation_time::types::ScheduledTasks<sp_core::crypto::AccountId32, primitive_types::H256>
     **/
    PalletAutomationTimeScheduledTasks: {
        tasks: 'Vec<(AccountId32,H256)>',
        weight: 'u128'
    },
    /**
     * Lookup453: pallet_automation_time::types::Task<sp_core::crypto::AccountId32, Balance, CurrencyId>
     **/
    PalletAutomationTimeTask: {
        ownerId: 'AccountId32',
        providedId: 'Bytes',
        schedule: 'PalletAutomationTimeSchedule',
        action: 'PalletAutomationTimeAction'
    },
    /**
     * Lookup454: pallet_automation_time::types::Schedule
     **/
    PalletAutomationTimeSchedule: {
        _enum: {
            Fixed: {
                executionTimes: 'Vec<u64>',
                executionsLeft: 'u32',
            },
            Recurring: {
                nextExecutionTime: 'u64',
                frequency: 'u64'
            }
        }
    },
    /**
     * Lookup455: pallet_automation_time::types::Action<sp_core::crypto::AccountId32, Balance, CurrencyId>
     **/
    PalletAutomationTimeAction: {
        _enum: {
            Notify: {
                message: 'Bytes',
            },
            NativeTransfer: {
                sender: 'AccountId32',
                recipient: 'AccountId32',
                amount: 'u128',
            },
            XCMP: {
                paraId: 'u32',
                currencyId: 'u32',
                encodedCall: 'Bytes',
                encodedCallWeight: 'u64',
            },
            AutoCompoundDelegatedStake: {
                delegator: 'AccountId32',
                collator: 'AccountId32',
                accountMinimum: 'u128',
            },
            DynamicDispatch: {
                encodedCall: 'Bytes'
            }
        }
    },
    /**
     * Lookup457: pallet_automation_time::types::MissedTaskV2<sp_core::crypto::AccountId32, primitive_types::H256>
     **/
    PalletAutomationTimeMissedTaskV2: {
        ownerId: 'AccountId32',
        taskId: 'H256',
        executionTime: 'u64'
    },
    /**
     * Lookup459: pallet_automation_time::pallet::Error<T>
     **/
    PalletAutomationTimeError: {
        _enum: ['InvalidTime', 'PastTime', 'TimeTooFarOut', 'EmptyMessage', 'EmptyProvidedId', 'DuplicateTask', 'TimeSlotFull', 'TaskDoesNotExist', 'BlockTimeNotSet', 'InvalidAmount', 'TransferToSelf', 'InsufficientBalance', 'LiquidityRestrictions', 'TooManyExecutionsTimes', 'CallCannotBeDecoded']
    },
    /**
     * Lookup461: pallet_vesting::pallet::Error<T>
     **/
    PalletVestingError: {
        _enum: ['BlockTimeNotSet']
    },
    /**
     * Lookup462: pallet_xcmp_handler::pallet::Error<T>
     **/
    PalletXcmpHandlerError: {
        _enum: ['CurrencyChainComboNotSupported', 'CurrencyChainComboNotFound', 'FeeOverflow', 'WeightOverflow', 'FailedMultiLocationToJunction', 'CannotReanchor', 'ErrorSendingXcmToTarget', 'XcmExecutionFailed', 'ErrorGettingCallWeight']
    },
    /**
     * Lookup466: pallet_automation_price::pallet::Task<T>
     **/
    PalletAutomationPriceTask: {
        ownerId: 'AccountId32',
        providedId: 'Bytes',
        asset: 'Bytes',
        direction: 'PalletAutomationPriceDirection',
        triggerPercentage: 'u128',
        action: 'PalletAutomationPriceAction'
    },
    /**
     * Lookup467: pallet_automation_price::pallet::Action<T>
     **/
    PalletAutomationPriceAction: {
        _enum: {
            NativeTransfer: {
                sender: 'AccountId32',
                recipient: 'AccountId32',
                amount: 'u128'
            }
        }
    },
    /**
     * Lookup469: pallet_automation_price::pallet::AssetMetadatum<T>
     **/
    PalletAutomationPriceAssetMetadatum: {
        upperBound: 'u16',
        lowerBound: 'u8',
        expirationPeriod: 'u64',
        assetSudo: 'AccountId32'
    },
    /**
     * Lookup470: pallet_automation_price::pallet::Error<T>
     **/
    PalletAutomationPriceError: {
        _enum: ['EmptyProvidedId', 'InvalidTime', 'DuplicateTask', 'AssetNotSupported', 'AssetAlreadySupported', 'InvalidAssetSudo', 'AssetNotInTriggerableRange', 'BlockTimeNotSet', 'InvalidAssetExpirationWindow', 'MaxTasksReached', 'TaskInsertionFailure', 'InsufficientBalance', 'LiquidityRestrictions', 'AssetLimitReached', 'DirectionNotSupported']
    },
    /**
     * Lookup472: sp_runtime::MultiSignature
     **/
    SpRuntimeMultiSignature: {
        _enum: {
            Ed25519: 'SpCoreEd25519Signature',
            Sr25519: 'SpCoreSr25519Signature',
            Ecdsa: 'SpCoreEcdsaSignature'
        }
    },
    /**
     * Lookup473: sp_core::ed25519::Signature
     **/
    SpCoreEd25519Signature: '[u8;64]',
    /**
     * Lookup475: sp_core::sr25519::Signature
     **/
    SpCoreSr25519Signature: '[u8;64]',
    /**
     * Lookup476: sp_core::ecdsa::Signature
     **/
    SpCoreEcdsaSignature: '[u8;65]',
    /**
     * Lookup479: frame_system::extensions::check_spec_version::CheckSpecVersion<T>
     **/
    FrameSystemExtensionsCheckSpecVersion: 'Null',
    /**
     * Lookup480: frame_system::extensions::check_tx_version::CheckTxVersion<T>
     **/
    FrameSystemExtensionsCheckTxVersion: 'Null',
    /**
     * Lookup481: frame_system::extensions::check_genesis::CheckGenesis<T>
     **/
    FrameSystemExtensionsCheckGenesis: 'Null',
    /**
     * Lookup484: frame_system::extensions::check_nonce::CheckNonce<T>
     **/
    FrameSystemExtensionsCheckNonce: 'Compact<u32>',
    /**
     * Lookup485: frame_system::extensions::check_weight::CheckWeight<T>
     **/
    FrameSystemExtensionsCheckWeight: 'Null',
    /**
     * Lookup486: pallet_transaction_payment::ChargeTransactionPayment<T>
     **/
    PalletTransactionPaymentChargeTransactionPayment: 'Compact<u128>',
    /**
     * Lookup487: turing_runtime::Runtime
     **/
    TuringRuntimeRuntime: 'Null'
};
